
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
        <meta name="author" content="Vishal Koparde, Ph.D.">
      
      
      
        <link rel="prev" href="../flowchart/">
      
      
        <link rel="next" href="../references/">
      
      
      <link rel="icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.5.3, mkdocs-material-9.5.2">
    
    
      
        <title>Tutorial - CHARLIE (Circrnas in Host And viRuses anaLysis pIpEline)</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.50c56a3b.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#charlie" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="CHARLIE (Circrnas in Host And viRuses anaLysis pIpEline)" class="md-header__button md-logo" aria-label="CHARLIE (Circrnas in Host And viRuses anaLysis pIpEline)" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            CHARLIE (Circrnas in Host And viRuses anaLysis pIpEline)
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Tutorial
            
          </span>
        </div>
      </div>
    </div>
    
      
    
    
    
    
      <label class="md-header__button md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/CCBR/CHARLIE" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.5.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    CCBR/CHARLIE
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="CHARLIE (Circrnas in Host And viRuses anaLysis pIpEline)" class="md-nav__button md-logo" aria-label="CHARLIE (Circrnas in Host And viRuses anaLysis pIpEline)" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    CHARLIE (Circrnas in Host And viRuses anaLysis pIpEline)
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/CCBR/CHARLIE" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.5.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    CCBR/CHARLIE
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Home
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../flowchart/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Flowchart
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  <span class="md-ellipsis">
    Tutorial
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  <span class="md-ellipsis">
    Tutorial
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#tutorial" class="md-nav__link">
    <span class="md-ellipsis">
      Tutorial
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Tutorial">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#prerequisites" class="md-nav__link">
    <span class="md-ellipsis">
      Prerequisites
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#location" class="md-nav__link">
    <span class="md-ellipsis">
      Location
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#init" class="md-nav__link">
    <span class="md-ellipsis">
      Init
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Init">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#configyaml" class="md-nav__link">
    <span class="md-ellipsis">
      config.yaml
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#samplestsv" class="md-nav__link">
    <span class="md-ellipsis">
      samples.tsv
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#dryrun" class="md-nav__link">
    <span class="md-ellipsis">
      Dryrun
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#run" class="md-nav__link">
    <span class="md-ellipsis">
      Run
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#runlocal" class="md-nav__link">
    <span class="md-ellipsis">
      Runlocal
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../references/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    References
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../versions/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Versions
  </span>
  

      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#tutorial" class="md-nav__link">
    <span class="md-ellipsis">
      Tutorial
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Tutorial">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#prerequisites" class="md-nav__link">
    <span class="md-ellipsis">
      Prerequisites
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#location" class="md-nav__link">
    <span class="md-ellipsis">
      Location
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#init" class="md-nav__link">
    <span class="md-ellipsis">
      Init
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Init">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#configyaml" class="md-nav__link">
    <span class="md-ellipsis">
      config.yaml
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#samplestsv" class="md-nav__link">
    <span class="md-ellipsis">
      samples.tsv
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#dryrun" class="md-nav__link">
    <span class="md-ellipsis">
      Dryrun
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#run" class="md-nav__link">
    <span class="md-ellipsis">
      Run
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#runlocal" class="md-nav__link">
    <span class="md-ellipsis">
      Runlocal
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  

  
  


<h1 id="charlie">CHARLIE</h1>
<p><img alt="img" src="https://img.shields.io/github/issues/kopardev/circRNA?style=for-the-badge" /><img alt="img" src="https://img.shields.io/github/forks/kopardev/circRNA?style=for-the-badge" /><img alt="img" src="https://img.shields.io/github/stars/kopardev/circRNA?style=for-the-badge" /><img alt="img" src="https://img.shields.io/github/license/kopardev/circRNA?style=for-the-badge" /></p>
<h3 id="tutorial">Tutorial</h3>
<h4 id="prerequisites">Prerequisites</h4>
<ul>
<li>
<p><a href="https://hpc.nih.gov/">Biowulf</a> account: Biowulf account can be requested <a href="https://hpc.nih.gov/docs/accounts.html">here</a></p>
</li>
<li>
<p>Membership to Ziegelbauer user group on Biowulf. You can check this by typing the following command:</p>
</li>
</ul>
<p><code>bash
  % groups</code></p>
<p>output:</p>
<p><code>bash
  CCBR kopardevn Ziegelbauer_lab</code></p>
<p>If <code>Ziegelbauer_lab</code> is not listed then you can email a request to be added to the groups <a href="mailto:staff@hpc.nih.gov">here</a></p>
<h4 id="location">Location</h4>
<p>Different versions of circRNA DAQ pipeline have been parked at <code>/data/Ziegelbauer_lab/Pipelines/circRNA</code></p>
<pre><code class="language-bash">% ls -la /data/Ziegelbauer_lab/Pipelines/circRNA
</code></pre>
<p>output:</p>
<pre><code class="language-bash">total 135
drwxrws--T. 6 kopardevn Ziegelbauer_lab 4096 Feb  4 18:27 .
drwxrws--T. 3 kopardevn Ziegelbauer_lab 4096 Jan 12 09:05 ..
lrwxrwxrwx. 1 kopardevn Ziegelbauer_lab   54 Feb  3 14:29 dev -&gt; /data/Ziegelbauer_lab/circRNADetection/scripts/circRNA
drwxrws---. 3 kopardevn Ziegelbauer_lab 4096 Jan 12 09:07 v0.1.0
drwxrws---. 6 kopardevn Ziegelbauer_lab 4096 Jan 12 17:32 v0.2.1
drwxrws---. 6 kopardevn Ziegelbauer_lab 4096 Jan 14 10:24 v0.3.3
drwxrws---. 7 kopardevn Ziegelbauer_lab 4096 Feb  4 18:26 v0.4.0
</code></pre>
<p>The exacts versions listed here may changed as newer versions are added. Also, the <code>dev</code> version is pointing to the most recent untagged version of the pipeline (use at own risk!)</p>
<h4 id="init">Init</h4>
<p>To get help about the pipeline you can run:</p>
<pre><code class="language-bash">% bash /data/Ziegelbauer_lab/Pipelines/circRNA/v0.4.0/run_circrna_daq.sh --help
</code></pre>
<p>output:</p>
<pre><code class="language-bash">Pipeline Dir: /data/Ziegelbauer_lab/circRNADetection/scripts/circRNA
Git Commit/Tag: b2f387c1b6854646d12974cd16da1168d93bb43b    v0.4.0-14-gb2f387c
run_circrna_daq.sh: run the workflow to DAQ (detect, annotate and quantify circRNAs)
USAGE:
  bash run_circrna_daq.sh &lt;MODE&gt;
Required Positional Argument:
  MODE: [Type: Str] Valid options:
    a) init &lt;path_to_workdir&gt; : initialize workdir
    b) run &lt;path_to_workdir&gt;: run with slurm
    c) reset &lt;path_to_workdir&gt; : DELETE workdir dir and re-init it
    e) dryrun &lt;path_to_workdir&gt; : dry run snakemake to generate DAG
    f) unlock &lt;path_to_workdir&gt; : unlock workdir if locked by snakemake
    g) runlocal &lt;path_to_workdir&gt;: run without submitting to sbatch
</code></pre>
<p>You can replace <code>v0.4.0</code> in the above command with the latest version to use a newer version. <code>run_circrna_daq.sh</code> was called <code>test.sh</code> in versions older than <code>v0.4.0</code>.</p>
<p>To initial the working directory run:</p>
<pre><code class="language-bash">% bash /data/Ziegelbauer_lab/Pipelines/circRNA/v0.4.0/run_circrna_daq.sh init /scratch/circRNA_daq_test
</code></pre>
<blockquote>
<p><strong>NOTE</strong></p>
<p>With version 0.6.0 or newer the arguments to the wrapper script need to be provided with -w/-m prefixes</p>
<p><code>bash
bash /data/Ziegelbauer_lab/Pipelines/circRNA/v0.6.0/run_circrna_daq.sh -m=init -w=/scratch/circRNA_daq_test</code></p>
</blockquote>
<p>This assumes that <code>/scratch/circRNA_daq_test</code> does not exist before running this command and is at a location where write permissions are available.</p>
<pre><code class="language-bash">% bash /data/Ziegelbauer_lab/Pipelines/circRNA/v0.4.0/run_circrna_daq.sh init /scratch/circRNA_daq_test
</code></pre>
<p>​   output:</p>
<pre><code class="language-bash">Pipeline Dir: /data/Ziegelbauer_lab/circRNADetection/scripts/circRNA
Git Commit/Tag: b2f387c1b6854646d12974cd16da1168d93bb43b    v0.4.0-14-gb2f387c
Working Dir: /scratch/circRNA_daq_test
/data/Ziegelbauer_lab/circRNADetection/scripts/circRNA
/scratch/circRNA_daq_test
Logs Dir: /scratch/circRNA_daq_test/logs
Stats Dir: /scratch/circRNA_daq_test/stats
Done Initializing /scratch/circRNA_daq_test. You can now edit /scratch/circRNA_daq_test/config.yaml and /scratch/circRNA_daq_test/samples.tsv
</code></pre>
<p>The above command creates <code>/scratch/circRNA_daq_test</code> folder and creates 2 subfolders <code>logs</code> and <code>stats</code> inside that folder along with <code>config.yaml</code> and <code>samples.tsv</code> files.</p>
<pre><code class="language-bash">% tree /scratch/circRNA_daq_test
</code></pre>
<p>​   output:</p>
<pre><code class="language-bash">/scratch/circRNA_daq_test
├── config.yaml
├── logs
├── samples.tsv
└── stats

2 directories, 2 files
</code></pre>
<h5 id="configyaml">config.yaml</h5>
<p>This file is used to fine tune the execution of the pipeline by setting:</p>
<ul>
<li>sample sheet location ... aka <code>samples.tsv</code></li>
<li>whether to run CLEAR pipeline or not by setting <strong>run_clear</strong> to <code>True</code> or <code>False</code></li>
<li>describes the location of other resources/indexes/tools etc. Generally, these do NOT need to be changed.</li>
</ul>
<h5 id="samplestsv">samples.tsv</h5>
<p>Tab delimited definition of sample sheet. The header is fixed and each row represents a sample. It has 3 columns:</p>
<ol>
<li>sampleName = Name of the sample. This has to be unique.</li>
<li>path_to_R1_fastq = absolute path to the read1 fastq.gz file.</li>
<li>path_to_R2_fastq = absolute path to the read2 fastq.gz file. If the sample was sequenced in single-end mode, then leave this blank.</li>
</ol>
<p>Running <strong>init</strong> will put the following example file in the workdir supplied and it looks like this:</p>
<pre><code class="language-bash">% more /scratch/circRNA_daq_test/samples.tsv
</code></pre>
<p>output:</p>
<pre><code class="language-bash">sampleName  path_to_R1_fastq    path_to_R2_fastq
GI1_N   /data/Ziegelbauer_lab/circRNADetection/rawdata/ccbr983/fastq2/5_GI112118_norm_S4_R1_001.fastq.gz    /data/Ziegelbauer_lab/circRNADetection/rawdata/ccbr983/fastq2/5_GI112118_norm_S4_R2_001.fastq.gz
GI1_T   /data/Ziegelbauer_lab/circRNADetection/rawdata/ccbr983/fastq2/6_GI112118_tum_S5_R1_001.fastq.gz
</code></pre>
<h4 id="dryrun">Dryrun</h4>
<p>Once the <code>samples.tsv</code> file has been edited appropriately to include the desired samples, it is a good idea to <strong>dryrun</strong> the pipeline to ensure that everything will work as desired. Dryrun can be run as follows:</p>
<pre><code class="language-bash">% bash /data/Ziegelbauer_lab/Pipelines/circRNA/v0.4.0/run_circrna_daq.sh dryrun /scratch/circRNA_daq_test
</code></pre>
<p><a href="../dryrun_example.txt">Here</a> is the output of the above command.</p>
<h4 id="run">Run</h4>
<p>Upon verifying that dryrun is successful. You can then submit the job to the cluster using the following command:</p>
<pre><code class="language-bash">% bash /data/Ziegelbauer_lab/Pipelines/circRNA/v0.4.0/run_circrna_daq.sh run /scratch/circRNA_daq_test
</code></pre>
<blockquote>
<p>NOTE:</p>
<p>With v0.6.0 or later the above command will be:</p>
<p><code>bash
bash /data/Ziegelbauer_lab/Pipelines/circRNA/v0.4.0/run_circrna_daq.sh -m=run -w=/scratch/circRNA_daq_test</code></p>
</blockquote>
<p>output:</p>
<pre><code class="language-bash">Pipeline Dir: /data/Ziegelbauer_lab/circRNADetection/scripts/circRNA
Git Commit/Tag: 37419bc0eb196fb1e137849ebeb2739a1c12126c    v0.4.0-16-g37419bc
Working Dir: /scratch/circRNA_daq_test
[+] Loading python 3.7  ...
[+] Loading snakemake  5.24.1
Running...
ls: cannot access /scratch/circRNA_daq_test/slurm-*.out: No such file or directory
7930909
</code></pre>
<p><code>7930909</code> is the jobid returned by the slurm job scheduler on biowulf. This means that the job was successfully submitted, it will spawn off other subjobs which in-turn will be run and outputs will be moved to the <code>results</code> folder created inside the working directory supplied at command line. You can check the status of your queue of jobs in biowulf running:</p>
<pre><code class="language-bash">% squeue -u `whoami`
</code></pre>
<p>output:</p>
<pre><code class="language-bash">             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           7930909  ccr,norm  circRNA kopardev PD       0:00      1 (None)
</code></pre>
<p><code>ST</code> in the above results stands for Status and <code>PD</code> means Pending. The status will change from pending(<code>PD</code>) to running(<code>R</code>) to completed as jobs are run on the cluster.</p>
<p>Next, just sit tight until the pipeline finishes. You can keep monitoring the queue as shown above. If there are no jobs running on biowulf, then your pipeline has finished (or errored out!)</p>
<p>Once completed the output should something like this:</p>
<pre><code class="language-bash">% tree /scratch/circRNA_daq_test
</code></pre>
<p>output:</p>
<pre><code class="language-bash">/scratch/circRNA_daq_test
├── config.yaml
├── fastqs
│   ├── GI1_N.R1.fastq.gz -&gt; /data/Ziegelbauer_lab/circRNADetection/rawdata/ccbr983/fastq2/5_GI112118_norm_S4_R1_001.fastq.gz
│   ├── GI1_N.R2.fastq.gz -&gt; /data/Ziegelbauer_lab/circRNADetection/rawdata/ccbr983/fastq2/5_GI112118_norm_S4_R2_001.fastq.gz
│   └── GI1_T.R1.fastq.gz -&gt; /data/Ziegelbauer_lab/circRNADetection/rawdata/ccbr983/fastq2/6_GI112118_tum_S5_R1_001.fastq.gz
├── logs
│   ├── 7658773.7658778.cutadapt.sample=GI1_T.err
│   ├── 7658773.7658778.cutadapt.sample=GI1_T.out
│   ├── 7658773.7658779.cutadapt.sample=GI1_N.err
│   ├── 7658773.7658779.cutadapt.sample=GI1_N.out
│   ├── 7658773.7659728.fastqc.sample=GI1_T.err
│   ├── 7658773.7659728.fastqc.sample=GI1_T.out
│   ├── 7658773.7659729.star1p.sample=GI1_T.err
│   ├── 7658773.7659729.star1p.sample=GI1_T.out
│   ├── 7658773.7659730.ciri.sample=GI1_T.err
│   ├── 7658773.7659730.ciri.sample=GI1_T.out
│   ├── 7658773.7659731.clear.sample=GI1_T.err
│   ├── 7658773.7659731.clear.sample=GI1_T.out
│   ├── 7658773.7660529.ciri.sample=GI1_N.err
│   ├── 7658773.7660529.ciri.sample=GI1_N.out
│   ├── 7658773.7660738.clear.sample=GI1_N.err
│   ├── 7658773.7660738.clear.sample=GI1_N.out
│   ├── 7658773.7660753.fastqc.sample=GI1_N.err
│   ├── 7658773.7660753.fastqc.sample=GI1_N.out
│   ├── 7658773.7660754.star1p.sample=GI1_N.err
│   ├── 7658773.7660754.star1p.sample=GI1_N.out
│   ├── 7658773.7661100.merge_SJ_tabs..err
│   ├── 7658773.7661100.merge_SJ_tabs..out
│   ├── 7658773.7661134.star2p.sample=GI1_T.err
│   ├── 7658773.7661134.star2p.sample=GI1_T.out
│   ├── 7658773.7661135.star2p.sample=GI1_N.err
│   ├── 7658773.7661135.star2p.sample=GI1_N.out
│   ├── 7658773.7662317.create_BSJ_bam.sample=GI1_T.err
│   ├── 7658773.7662317.create_BSJ_bam.sample=GI1_T.out
│   ├── 7658773.7662319.annotate_circRNA.sample=GI1_T.err
│   ├── 7658773.7662319.annotate_circRNA.sample=GI1_T.out
│   ├── 7658773.7662563.annotate_circRNA.sample=GI1_N.err
│   ├── 7658773.7662563.annotate_circRNA.sample=GI1_N.out
│   ├── 7658773.7662564.create_BSJ_bam.sample=GI1_N.err
│   ├── 7658773.7662564.create_BSJ_bam.sample=GI1_N.out
│   ├── 7658773.7662567.split_BAM_create_BW.sample=GI1_T.err
│   ├── 7658773.7662567.split_BAM_create_BW.sample=GI1_T.out
│   ├── 7658773.7663000.create_circexplorer_count_matrix..err
│   ├── 7658773.7663000.create_circexplorer_count_matrix..out
│   ├── 7658773.7663642.create_ciri_count_matrix..err
│   ├── 7658773.7663642.create_ciri_count_matrix..out
│   ├── 7658773.7663643.split_BAM_create_BW.sample=GI1_N.err
│   ├── 7658773.7663643.split_BAM_create_BW.sample=GI1_N.out
│   └── slurm-7658773.out.gz
├── qc
│   └── fastqc
│       ├── GI1_N.R1_fastqc.html
│       ├── GI1_N.R1_fastqc.zip
│       ├── GI1_N.R1.trim_fastqc.html
│       ├── GI1_N.R1.trim_fastqc.zip
│       ├── GI1_N.R2_fastqc.html
│       ├── GI1_N.R2_fastqc.zip
│       ├── GI1_N.R2.trim_fastqc.html
│       ├── GI1_N.R2.trim_fastqc.zip
│       ├── GI1_T.R1_fastqc.html
│       ├── GI1_T.R1_fastqc.zip
│       ├── GI1_T.R1.trim_fastqc.html
│       └── GI1_T.R1.trim_fastqc.zip
├── report.html
├── results
│   ├── circExplorer_BSJ_count_matrix.txt
│   ├── circExplorer_BSJ_count_matrix_with_annotations.txt
│   ├── circExplorer_count_matrix.txt
│   ├── circExplorer_count_matrix_with_annotations.txt
│   ├── ciri_count_matrix.txt
│   ├── ciri_count_matrix_with_annotations.txt
│   ├── GI1_N
│   │   ├── circExplorer
│   │   │   ├── GI1_N.back_spliced_junction.bed
│   │   │   ├── GI1_N_circexplorer_parse.log
│   │   │   ├── GI1_N.circularRNA_known.txt
│   │   │   └── low_conf_GI1_N.circularRNA_known.txt
│   │   ├── ciri
│   │   │   ├── CIRIerror.log
│   │   │   ├── GI1_N.bwa.bam
│   │   │   ├── GI1_N.bwa.log
│   │   │   ├── GI1_N.ciri.log
│   │   │   └── GI1_N.ciri.out
│   │   ├── CLEAR
│   │   │   ├── circ
│   │   │   │   └── bsj.bed
│   │   │   ├── fusion
│   │   │   │   └── junctions.bed
│   │   │   ├── hisat
│   │   │   │   └── sp.txt
│   │   │   └── quant
│   │   │       └── quant.txt
│   │   ├── STAR1p
│   │   │   ├── GI1_N_p1.Chimeric.out.junction
│   │   │   ├── GI1_N_p1.Log.final.out
│   │   │   ├── GI1_N_p1.Log.out
│   │   │   ├── GI1_N_p1.Log.progress.out
│   │   │   └── GI1_N_p1.SJ.out.tab
│   │   ├── STAR2p
│   │   │   ├── GI1_N.BSJ.bam
│   │   │   ├── GI1_N.BSJ.bam.bai
│   │   │   ├── GI1_N.BSJ.ERCC.bam
│   │   │   ├── GI1_N.BSJ.ERCC.bam.bai
│   │   │   ├── GI1_N.BSJ.ERCC.bw
│   │   │   ├── GI1_N.BSJ.hg38.bam
│   │   │   ├── GI1_N.BSJ.hg38.bam.bai
│   │   │   ├── GI1_N.BSJ.hg38.bw
│   │   │   ├── GI1_N.BSJ.MN485971.1.bam
│   │   │   ├── GI1_N.BSJ.MN485971.1.bam.bai
│   │   │   ├── GI1_N.BSJ.MN485971.1.bw
│   │   │   ├── GI1_N.BSJ.NC_000898.1.bam
│   │   │   ├── GI1_N.BSJ.NC_000898.1.bam.bai
│   │   │   ├── GI1_N.BSJ.NC_000898.1.bw
│   │   │   ├── GI1_N.BSJ.NC_001664.4.bam
│   │   │   ├── GI1_N.BSJ.NC_001664.4.bam.bai
│   │   │   ├── GI1_N.BSJ.NC_001664.4.bw
│   │   │   ├── GI1_N.BSJ.NC_001716.2.bam
│   │   │   ├── GI1_N.BSJ.NC_001716.2.bam.bai
│   │   │   ├── GI1_N.BSJ.NC_001716.2.bw
│   │   │   ├── GI1_N.BSJ.NC_006273.2.bam
│   │   │   ├── GI1_N.BSJ.NC_006273.2.bam.bai
│   │   │   ├── GI1_N.BSJ.NC_006273.2.bw
│   │   │   ├── GI1_N.BSJ.NC_007605.1.bam
│   │   │   ├── GI1_N.BSJ.NC_007605.1.bam.bai
│   │   │   ├── GI1_N.BSJ.NC_007605.1.bw
│   │   │   ├── GI1_N.BSJ.NC_009333.1.bam
│   │   │   ├── GI1_N.BSJ.NC_009333.1.bam.bai
│   │   │   ├── GI1_N.BSJ.NC_009333.1.bw
│   │   │   ├── GI1_N.BSJ.NC_045512.2.bam
│   │   │   ├── GI1_N.BSJ.NC_045512.2.bam.bai
│   │   │   ├── GI1_N.BSJ.NC_045512.2.bw
│   │   │   ├── GI1_N.BSJ.readids
│   │   │   ├── GI1_N.BSJ.rRNA.bam
│   │   │   ├── GI1_N.BSJ.rRNA.bam.bai
│   │   │   ├── GI1_N.BSJ.rRNA.bw
│   │   │   ├── GI1_N_p2.Aligned.sortedByCoord.out.bam
│   │   │   ├── GI1_N_p2.Aligned.sortedByCoord.out.bam.bai
│   │   │   ├── GI1_N_p2.Chimeric.out.junction
│   │   │   ├── GI1_N_p2.Log.final.out
│   │   │   ├── GI1_N_p2.Log.out
│   │   │   ├── GI1_N_p2.Log.progress.out
│   │   │   └── GI1_N_p2.SJ.out.tab
│   │   └── trim
│   │       ├── GI1_N.R1.trim.fastq.gz
│   │       └── GI1_N.R2.trim.fastq.gz
│   ├── GI1_T
│   │   ├── circExplorer
│   │   │   ├── GI1_T.back_spliced_junction.bed
│   │   │   ├── GI1_T_circexplorer_parse.log
│   │   │   ├── GI1_T.circularRNA_known.txt
│   │   │   └── low_conf_GI1_T.circularRNA_known.txt
│   │   ├── ciri
│   │   │   ├── CIRIerror.log
│   │   │   ├── GI1_T.bwa.bam
│   │   │   ├── GI1_T.bwa.log
│   │   │   ├── GI1_T.ciri.log
│   │   │   └── GI1_T.ciri.out
│   │   ├── CLEAR
│   │   │   ├── circ
│   │   │   │   └── bsj.bed
│   │   │   ├── fusion
│   │   │   │   └── junctions.bed
│   │   │   ├── hisat
│   │   │   │   └── sp.txt
│   │   │   └── quant
│   │   │       └── quant.txt
│   │   ├── STAR1p
│   │   │   ├── GI1_T_p1.Chimeric.out.junction
│   │   │   ├── GI1_T_p1.Log.final.out
│   │   │   ├── GI1_T_p1.Log.out
│   │   │   ├── GI1_T_p1.Log.progress.out
│   │   │   └── GI1_T_p1.SJ.out.tab
│   │   ├── STAR2p
│   │   │   ├── GI1_T.BSJ.bam
│   │   │   ├── GI1_T.BSJ.bam.bai
│   │   │   ├── GI1_T.BSJ.ERCC.bam
│   │   │   ├── GI1_T.BSJ.ERCC.bam.bai
│   │   │   ├── GI1_T.BSJ.ERCC.bw
│   │   │   ├── GI1_T.BSJ.hg38.bam
│   │   │   ├── GI1_T.BSJ.hg38.bam.bai
│   │   │   ├── GI1_T.BSJ.hg38.bw
│   │   │   ├── GI1_T.BSJ.MN485971.1.bam
│   │   │   ├── GI1_T.BSJ.MN485971.1.bam.bai
│   │   │   ├── GI1_T.BSJ.MN485971.1.bw
│   │   │   ├── GI1_T.BSJ.NC_000898.1.bam
│   │   │   ├── GI1_T.BSJ.NC_000898.1.bam.bai
│   │   │   ├── GI1_T.BSJ.NC_000898.1.bw
│   │   │   ├── GI1_T.BSJ.NC_001664.4.bam
│   │   │   ├── GI1_T.BSJ.NC_001664.4.bam.bai
│   │   │   ├── GI1_T.BSJ.NC_001664.4.bw
│   │   │   ├── GI1_T.BSJ.NC_001716.2.bam
│   │   │   ├── GI1_T.BSJ.NC_001716.2.bam.bai
│   │   │   ├── GI1_T.BSJ.NC_001716.2.bw
│   │   │   ├── GI1_T.BSJ.NC_006273.2.bam
│   │   │   ├── GI1_T.BSJ.NC_006273.2.bam.bai
│   │   │   ├── GI1_T.BSJ.NC_006273.2.bw
│   │   │   ├── GI1_T.BSJ.NC_007605.1.bam
│   │   │   ├── GI1_T.BSJ.NC_007605.1.bam.bai
│   │   │   ├── GI1_T.BSJ.NC_007605.1.bw
│   │   │   ├── GI1_T.BSJ.NC_009333.1.bam
│   │   │   ├── GI1_T.BSJ.NC_009333.1.bam.bai
│   │   │   ├── GI1_T.BSJ.NC_009333.1.bw
│   │   │   ├── GI1_T.BSJ.NC_045512.2.bam
│   │   │   ├── GI1_T.BSJ.NC_045512.2.bam.bai
│   │   │   ├── GI1_T.BSJ.NC_045512.2.bw
│   │   │   ├── GI1_T.BSJ.readids
│   │   │   ├── GI1_T.BSJ.rRNA.bam
│   │   │   ├── GI1_T.BSJ.rRNA.bam.bai
│   │   │   ├── GI1_T.BSJ.rRNA.bw
│   │   │   ├── GI1_T_p2.Aligned.sortedByCoord.out.bam
│   │   │   ├── GI1_T_p2.Aligned.sortedByCoord.out.bam.bai
│   │   │   ├── GI1_T_p2.Chimeric.out.junction
│   │   │   ├── GI1_T_p2.Log.final.out
│   │   │   ├── GI1_T_p2.Log.out
│   │   │   ├── GI1_T_p2.Log.progress.out
│   │   │   └── GI1_T_p2.SJ.out.tab
│   │   └── trim
│   │       ├── GI1_T.R1.trim.fastq.gz
│   │       └── GI1_T.R2.trim.fastq.gz
│   └── pass1.out.tab
├── samples.tsv
├── stats
│   ├── snakemake.log.20210204103112.gz
│   └── snakemake.stats.20210204103112.gz
├── submit_script.sbatch
</code></pre>
<h4 id="runlocal">Runlocal</h4>
<p>If you have grabbed an interactive node using <code>sinteractive</code> and have a small test dataset in the <code>samples.tsv</code> and simply want to quickly check if everything works as expected. You can run <strong>locally</strong>, i.e. directly on the interactive node without submitting to the cluster using:</p>
<pre><code class="language-bash">% bash /data/Ziegelbauer_lab/Pipelines/circRNA/v0.4.0/run_circrna_daq.sh runlocal /scratch/circRNA_daq_test
</code></pre>
<p>This is only for testing purposes, do not use it for running 10s of samples as you will be soon timed out of the interactive node.</p>












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      Copyright &copy; 2023 CCBR
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
        <div class="md-social">
  
    
    
    
    
      
      
    
    <a href="http://bioinformatics.cancer.gov" target="_blank" rel="noopener" title="bioinformatics.cancer.gov" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 640 512"><!--! Font Awesome Free 6.5.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M144 0a80 80 0 1 1 0 160 80 80 0 1 1 0-160zm368 0a80 80 0 1 1 0 160 80 80 0 1 1 0-160zM0 298.7C0 239.8 47.8 192 106.7 192h42.7c15.9 0 31 3.5 44.6 9.7-1.3 7.2-1.9 14.7-1.9 22.3 0 38.2 16.8 72.5 43.3 96H21.3C9.6 320 0 310.4 0 298.7zM405.3 320h-.7c26.6-23.5 43.3-57.8 43.3-96 0-7.6-.7-15-1.9-22.3 13.6-6.3 28.7-9.7 44.6-9.7h42.7c58.9 0 106.7 47.8 106.7 106.7 0 11.8-9.6 21.3-21.3 21.3H405.3zM224 224a96 96 0 1 1 192 0 96 96 0 1 1-192 0zm-96 261.3c0-73.6 59.7-133.3 133.3-133.3h117.4c73.6 0 133.3 59.7 133.3 133.3 0 14.7-11.9 26.7-26.7 26.7H154.7c-14.7 0-26.7-11.9-26.7-26.7z"/></svg>
    </a>
  
    
    
    
    
      
      
    
    <a href="https://github.com/CCBR" target="_blank" rel="noopener" title="github.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.5.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg>
    </a>
  
    
    
    
    
      
      
    
    <a href="https://hub.docker.com/orgs/nciccbr/repositories" target="_blank" rel="noopener" title="hub.docker.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 640 512"><!--! Font Awesome Free 6.5.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M349.9 236.3h-66.1v-59.4h66.1v59.4zm0-204.3h-66.1v60.7h66.1V32zm78.2 144.8H362v59.4h66.1v-59.4zm-156.3-72.1h-66.1v60.1h66.1v-60.1zm78.1 0h-66.1v60.1h66.1v-60.1zm276.8 100c-14.4-9.7-47.6-13.2-73.1-8.4-3.3-24-16.7-44.9-41.1-63.7l-14-9.3-9.3 14c-18.4 27.8-23.4 73.6-3.7 103.8-8.7 4.7-25.8 11.1-48.4 10.7H2.4c-8.7 50.8 5.8 116.8 44 162.1 37.1 43.9 92.7 66.2 165.4 66.2 157.4 0 273.9-72.5 328.4-204.2 21.4.4 67.6.1 91.3-45.2 1.5-2.5 6.6-13.2 8.5-17.1l-13.3-8.9zm-511.1-27.9h-66v59.4h66.1v-59.4zm78.1 0h-66.1v59.4h66.1v-59.4zm78.1 0h-66.1v59.4h66.1v-59.4zm-78.1-72.1h-66.1v60.1h66.1v-60.1z"/></svg>
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    <script id="__config" type="application/json">{"base": "..", "features": [], "search": "../assets/javascripts/workers/search.f886a092.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../assets/javascripts/bundle.d7c377c4.min.js"></script>
      
    
  </body>
</html>