<!DOCTYPE html><html class=writer-html5 lang=en> <head><meta charset=utf-8><meta http-equiv=X-UA-Compatible content="IE=edge"><meta name=viewport content="width=device-width, initial-scale=1.0"><meta name=description content=None><meta name=author content="Vishal Koparde, Ph.D."><link rel="shortcut icon" href=img/favicon.ico><title>CHARLIE (Circrnas in Host And viRuses anaLysis pIpEline)</title><link rel=stylesheet href=css/theme.css><link rel=stylesheet href=css/theme_extra.css><link rel=stylesheet href=https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/styles/github.min.css><link href=css/custom.css rel=stylesheet><link href=css/version-select.css rel=stylesheet><script>
        // Current page data
        var mkdocs_page_name = "Home";
        var mkdocs_page_input_path = "index.md";
        var mkdocs_page_url = null;
      </script><!--[if lt IE 9]>
      <script src="js/html5shiv.min.js"></script>
    <![endif]--><script src=https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/highlight.min.js></script><script>hljs.highlightAll();</script></head> <body class=wy-body-for-nav role=document> <div class=wy-grid-for-nav> <nav data-toggle=wy-nav-shift class="wy-nav-side stickynav"> <div class=wy-side-scroll> <div class=wy-side-nav-search> <a href=. class="icon icon-home"> CHARLIE (Circrnas in Host And viRuses anaLysis pIpEline) </a><div role=search> <form id=rtd-search-form class=wy-form action=./search.html method=get> <input type=text name=q placeholder="Search docs" aria-label="Search docs" title="Type search term here"> </form> </div> </div> <div class="wy-menu wy-menu-vertical" data-spy=affix role=navigation aria-label="Navigation menu"> <ul class=current> <li class="toctree-l1 current"><a class="reference internal current" href=#>Home</a> <ul class=current> <li class=toctree-l2><a class="reference internal" href=#table-of-contents>Table of Contents</a> </li> <li class=toctree-l2><a class="reference internal" href=#1-introduction>1. Introduction</a> </li> <li class=toctree-l2><a class="reference internal" href=#2-flowchart>2. Flowchart</a> </li> <li class=toctree-l2><a class="reference internal" href=#3-software-dependencies>3. Software Dependencies</a> </li> <li class=toctree-l2><a class="reference internal" href=#4-usage>4. Usage</a> </li> <li class=toctree-l2><a class="reference internal" href=#5-license>5. License</a> </li> <li class=toctree-l2><a class="reference internal" href=#6-testing>6. Testing</a> <ul> <li class=toctree-l3><a class="reference internal" href=#init>Init</a> </li> <li class=toctree-l3><a class="reference internal" href=#dry-run>Dry-run</a> </li> <li class=toctree-l3><a class="reference internal" href=#run>Run</a> <ul> <li class=toctree-l4><a class="reference internal" href=#61-test-data>6.1 Test Data</a> </li> <li class=toctree-l4><a class="reference internal" href=#62-expected-output>6.2 Expected Output</a> </li> </ul> </li> </ul> </li> </ul> </li> </ul> <ul> <li class=toctree-l1><a href=flowchart/ class="reference internal">Workflow</a> </li> </ul> <ul> <li class=toctree-l1><a href=tutorial/ class="reference internal">Tutorial</a> </li> </ul> <ul> <li class=toctree-l1><a href=references/ class="reference internal">References</a> </li> </ul> <ul> <li class=toctree-l1><a href=CHANGELOG/ class="reference internal">Changelog</a> </li> </ul> <ul> <li class=toctree-l1><a href=contributing/ class="reference internal">How to contribute</a> </li> </ul> </div> </div> </nav> <section data-toggle=wy-nav-shift class=wy-nav-content-wrap> <nav class=wy-nav-top role=navigation aria-label="Mobile navigation menu"> <i data-toggle=wy-nav-top class="fa fa-bars"></i> <a href=.>CHARLIE (Circrnas in Host And viRuses anaLysis pIpEline)</a> </nav> <div class=wy-nav-content> <div class=rst-content><div role=navigation aria-label="breadcrumbs navigation"> <ul class=wy-breadcrumbs> <li><a href=. class="icon icon-home" aria-label=Docs></a></li> <li class="breadcrumb-item active">Home</li> <li class=wy-breadcrumbs-aside> <a href=https://github.com/CCBR/CHARLIE/edit/main/index.md>Edit on CCBR/CHARLIE</a> </li> </ul> <hr> </div> <div role=main class=document itemscope=itemscope itemtype=http://schema.org/Article> <div class=section itemprop=articleBody> <h1 id=charlie>CHARLIE<a class=headerlink href=#charlie title="Permanent link">&para;</a></h1> <p><strong>C</strong>ircrnas in <strong>H</strong>ost <strong>A</strong>nd vi<strong>R</strong>uses ana<strong>L</strong>ysis p<strong>I</strong>p<strong>E</strong>line</p> <p><a href=https://github.com/CCBR/CHARLIE/actions/workflows/build.yml><img alt=build src=https://github.com/CCBR/CHARLIE/actions/workflows/build.yml/badge.svg></a> <a href=https://ccbr.github.io/CHARLIE><img alt=docs src=https://github.com/CCBR/CHARLIE/actions/workflows/docs-mkdocs.yml/badge.svg></a> <a href=https://github.com/CCBR/CHARLIE/blob/main/LICENSE><img alt=license src=https://img.shields.io/github/license/CCBR/CHARLIE></a> <a href=https://github.com/CCBR/CHARLIE/issues><img alt=issues src=https://img.shields.io/github/issues/CCBR/CHARLIE></a> <a href=https://github.com/CCBR/CHARLIE/pulls><img alt=PRs src=https://img.shields.io/github/issues-pr/CCBR/CHARLIE></a> <img alt=forks src=https://img.shields.io/github/forks/CCBR/CHARLIE> <img alt=stars src=https://img.shields.io/github/stars/CCBR/CHARLIE> <a href=https://doi.org/10.5281/zenodo.13770166><img alt=DOI src=https://zenodo.org/badge/DOI/10.5281/zenodo.13770166.svg></a></p> <p>See the website for detailed information, documentation, and examples: <a href=https://ccbr.github.io/CHARLIE/ >https://ccbr.github.io/CHARLIE/</a></p> <h3 id=table-of-contents>Table of Contents<a class=headerlink href=#table-of-contents title="Permanent link">&para;</a></h3> <ul> <li><a href=#charlie>CHARLIE</a></li> <li><a href=#table-of-contents>Table of Contents</a></li> <li><a href=#1-introduction>1. Introduction</a></li> <li><a href=#2-flowchart>2. Flowchart</a></li> <li><a href=#3-software-dependencies>3. Software Dependencies</a></li> <li><a href=#4-usage>4. Usage</a></li> <li><a href=#5-license>5. License</a></li> <li><a href=#6-testing>6. Testing</a><ul> <li><a href=#61-test-data>6.1 Test data</a></li> <li><a href=#62-expected-output>6.2 Expected output</a></li> </ul> </li> </ul> <h3 id=1-introduction>1. Introduction<a class=headerlink href=#1-introduction title="Permanent link">&para;</a></h3> <p><strong>C</strong>ircrnas in <strong>H</strong>ost <strong>A</strong>nd vi<strong>R</strong>uses ana<strong>L</strong>ysis p<strong>I</strong>p<strong>E</strong>line</p> <p>Things to know about CHARLIE:</p> <ul> <li>Snakemake workflow to detect, annotate and quantify (DAQ) host and viral circular RNAs.</li> <li>Primirarily developed to run on <a href=https://hpc.nih.gov/ >BIOWULF</a></li> <li>Reach out to <a href=mailto:vishal.koparde@nihgov>Vishal Koparde</a> for questions/comments/requests.</li> </ul> <p>This circularRNA detection pipeline uses CIRCExplorer2, CIRI2 and many other tools in parallel to detect, quantify and annotate circRNAs. Here is a list of tools that can be run using CHARLIE:</p> <table> <thead> <tr> <th>circRNA Detection Tool</th> <th>Aligner(s)</th> <th>Run by default</th> </tr> </thead> <tbody> <tr> <td><a href=https://github.com/YangLab/CIRCexplorer2>CIRCExplorer2</a></td> <td>STAR<sup>1</sup></td> <td>Yes</td> </tr> <tr> <td><a href=https://sourceforge.net/projects/ciri/files/CIRI2/ >CIRI2</a></td> <td>BWA<sup>1</sup></td> <td>Yes</td> </tr> <tr> <td><a href=https://github.com/YangLab/CIRCexplorer2>CIRCExplorer2</a></td> <td>BWA<sup>1</sup></td> <td>Yes</td> </tr> <tr> <td><a href=https://github.com/YangLab/CLEAR>CLEAR</a></td> <td>STAR<sup>1</sup></td> <td>Yes</td> </tr> <tr> <td><a href=https://github.com/dieterich-lab/DCC>DCC</a></td> <td>STAR<sup>2</sup></td> <td>Yes</td> </tr> <tr> <td><a href=https://github.com/bioxfu/circRNAFinder>circRNAFinder</a></td> <td>STAR<sup>3</sup></td> <td>Yes</td> </tr> <tr> <td><a href=https://github.com/marvin-jens/find_circ>find_circ</a></td> <td>Bowtie2</td> <td>Yes</td> </tr> <tr> <td><a href=https://github.com/merckey/MapSplice2>MapSplice</a></td> <td>BWA<sup>2</sup></td> <td>No</td> </tr> <tr> <td><a href=https://github.com/TreesLab/NCLscan>NCLScan</a></td> <td>NovoAlign</td> <td>No</td> </tr> </tbody> </table> <blockquote> <p>Note: STAR<sup>1</sup>, STAR<sup>2</sup>, STAR<sup>3</sup> denote 3 different sets of alignment parameters, etc.</p> <p>Note: BWA<sup>1</sup>, BWA<sup>2</sup> denote 2 different alignment parameters, etc.</p> </blockquote> <h3 id=2-flowchart>2. Flowchart<a class=headerlink href=#2-flowchart title="Permanent link">&para;</a></h3> <p><img alt=flowchart src=https://raw.githubusercontent.com/CCBR/CHARLIE/main/docs/images/CHARLIE_v0.8.x.png></p> <p>For complete documentation, view the website <a href=https://CCBR.github.io/CHARLIE/ >https://CCBR.github.io/CHARLIE/</a>.</p> <blockquote> <p>⚠️ DISCLAIMER: New circRNA tools have been added CHARLIE and the documentation is currently out of date!</p> </blockquote> <h3 id=3-software-dependencies>3. Software Dependencies<a class=headerlink href=#3-software-dependencies title="Permanent link">&para;</a></h3> <p>CHARLIE is already installed on biowulf. It is included in the ccbrpipeliner module from release 7 onward. To load the module run:</p> <div class=highlight><pre><span></span><code>module<span class=w> </span>load<span class=w> </span>ccbrpipeliner/7
</code></pre></div> <p>The following version of various bioinformatics tools are using within CHARLIE:</p> <table> <thead> <tr> <th>tool</th> <th>version</th> </tr> </thead> <tbody> <tr> <td>blat</td> <td>3.5</td> </tr> <tr> <td>bedtools</td> <td>2.30.0</td> </tr> <tr> <td>bowtie</td> <td>2-2.5.1</td> </tr> <tr> <td>bowtie</td> <td>1.3.1</td> </tr> <tr> <td>bwa</td> <td>0.7.17</td> </tr> <tr> <td>circexplorer2</td> <td>2.3.8</td> </tr> <tr> <td>cufflinks</td> <td>2.2.1</td> </tr> <tr> <td>cutadapt</td> <td>4.4</td> </tr> <tr> <td>fastqc</td> <td>0.11.9</td> </tr> <tr> <td>hisat</td> <td>2.2.2.1</td> </tr> <tr> <td>java</td> <td>18.0.1.1</td> </tr> <tr> <td>multiqc</td> <td>1.9</td> </tr> <tr> <td>parallel</td> <td>20231122</td> </tr> <tr> <td>perl</td> <td>5.34</td> </tr> <tr> <td>picard</td> <td>2.27.3</td> </tr> <tr> <td>python</td> <td>2.7</td> </tr> <tr> <td>python</td> <td>3.8</td> </tr> <tr> <td>sambamba</td> <td>0.8.2</td> </tr> <tr> <td>samtools</td> <td>1.16.1</td> </tr> <tr> <td>STAR</td> <td>2.7.6a</td> </tr> <tr> <td>stringtie</td> <td>2.2.1</td> </tr> <tr> <td>ucsc</td> <td>450</td> </tr> <tr> <td>R</td> <td>4.0.5</td> </tr> <tr> <td>novocraft</td> <td>4.03.05</td> </tr> </tbody> </table> <h3 id=4-usage>4. Usage<a class=headerlink href=#4-usage title="Permanent link">&para;</a></h3> <div class=highlight><pre><span></span><code>charlie


<span class=c1>##########################################################################################</span>

Welcome<span class=w> </span>to
<span class=w> </span>_______<span class=w>  </span>__<span class=w>   </span>__<span class=w>  </span>_______<span class=w>  </span>______<span class=w>    </span>___<span class=w>      </span>___<span class=w>   </span>_______
<span class=p>|</span><span class=w>       </span><span class=o>||</span><span class=w>  </span><span class=p>|</span><span class=w> </span><span class=p>|</span><span class=w>  </span><span class=o>||</span><span class=w>   </span>_<span class=w>   </span><span class=o>||</span><span class=w>    </span>_<span class=w> </span><span class=p>|</span><span class=w>  </span><span class=p>|</span><span class=w>   </span><span class=p>|</span><span class=w>    </span><span class=p>|</span><span class=w>   </span><span class=p>|</span><span class=w> </span><span class=p>|</span><span class=w>       </span><span class=p>|</span>
<span class=p>|</span><span class=w>       </span><span class=o>||</span><span class=w>  </span><span class=p>|</span>_<span class=p>|</span><span class=w>  </span><span class=o>||</span><span class=w>  </span><span class=p>|</span>_<span class=p>|</span><span class=w>  </span><span class=o>||</span><span class=w>   </span><span class=p>|</span><span class=w> </span><span class=o>||</span><span class=w>  </span><span class=p>|</span><span class=w>   </span><span class=p>|</span><span class=w>    </span><span class=p>|</span><span class=w>   </span><span class=p>|</span><span class=w> </span><span class=p>|</span><span class=w>    </span>___<span class=p>|</span>
<span class=p>|</span><span class=w>       </span><span class=o>||</span><span class=w>       </span><span class=o>||</span><span class=w>       </span><span class=o>||</span><span class=w>   </span><span class=p>|</span>_<span class=o>||</span>_<span class=w> </span><span class=p>|</span><span class=w>   </span><span class=p>|</span><span class=w>    </span><span class=p>|</span><span class=w>   </span><span class=p>|</span><span class=w> </span><span class=p>|</span><span class=w>   </span><span class=p>|</span>___
<span class=p>|</span><span class=w>      </span>_<span class=o>||</span><span class=w>       </span><span class=o>||</span><span class=w>       </span><span class=o>||</span><span class=w>    </span>__<span class=w>  </span><span class=o>||</span><span class=w>   </span><span class=p>|</span>___<span class=w> </span><span class=p>|</span><span class=w>   </span><span class=p>|</span><span class=w> </span><span class=p>|</span><span class=w>    </span>___<span class=p>|</span>
<span class=p>|</span><span class=w>     </span><span class=p>|</span>_<span class=w> </span><span class=p>|</span><span class=w>   </span>_<span class=w>   </span><span class=o>||</span><span class=w>   </span>_<span class=w>   </span><span class=o>||</span><span class=w>   </span><span class=p>|</span><span class=w>  </span><span class=p>|</span><span class=w> </span><span class=o>||</span><span class=w>       </span><span class=o>||</span><span class=w>   </span><span class=p>|</span><span class=w> </span><span class=p>|</span><span class=w>   </span><span class=p>|</span>___
<span class=p>|</span>_______<span class=o>||</span>__<span class=p>|</span><span class=w> </span><span class=p>|</span>__<span class=o>||</span>__<span class=p>|</span><span class=w> </span><span class=p>|</span>__<span class=o>||</span>___<span class=p>|</span><span class=w>  </span><span class=p>|</span>_<span class=o>||</span>_______<span class=o>||</span>___<span class=p>|</span><span class=w> </span><span class=p>|</span>_______<span class=p>|</span>

C_ircrnas<span class=w> </span><span class=k>in</span><span class=w> </span>H_ost<span class=w> </span>A_nd<span class=w> </span>vi_R_uses<span class=w> </span>ana_L_ysis<span class=w> </span>p_I_p_E_line

<span class=c1>##########################################################################################</span>

This<span class=w> </span>pipeline<span class=w> </span>was<span class=w> </span>built<span class=w> </span>by<span class=w> </span>CCBR<span class=w> </span><span class=o>(</span>https://bioinformatics.ccr.cancer.gov/ccbr<span class=o>)</span>
Please<span class=w> </span>contact<span class=w> </span>Vishal<span class=w> </span>Koparde<span class=w> </span><span class=k>for</span><span class=w> </span>comments/questions<span class=w> </span><span class=o>(</span>vishal.koparde@nih.gov<span class=o>)</span>

<span class=c1>##########################################################################################</span>

CHARLIE<span class=w> </span>can<span class=w> </span>be<span class=w> </span>used<span class=w> </span>to<span class=w> </span>DAQ<span class=o>(</span>Detect/Annotate/Quantify<span class=o>)</span><span class=w> </span>circRNAs<span class=w> </span><span class=k>in</span><span class=w> </span>hosts<span class=w> </span>and<span class=w> </span>viruses.

Here<span class=w> </span>is<span class=w> </span>the<span class=w> </span>list<span class=w> </span>of<span class=w> </span>hosts<span class=w> </span>and<span class=w> </span>viruses<span class=w> </span>that<span class=w> </span>are<span class=w> </span>currently<span class=w> </span>supported:

HOSTS:
<span class=w>  </span>*<span class=w> </span>hg38<span class=w>          </span><span class=o>[</span>Human<span class=o>]</span>
<span class=w>  </span>*<span class=w> </span>mm39<span class=w>          </span><span class=o>[</span>Mouse<span class=o>]</span>

ADDITIVES:
<span class=w>  </span>*<span class=w> </span>ERCC<span class=w>          </span><span class=o>[</span>External<span class=w> </span>RNA<span class=w> </span>Control<span class=w> </span>Consortium<span class=w> </span>sequences<span class=o>]</span>
<span class=w>  </span>*<span class=w> </span>BAC16Insert<span class=w>   </span><span class=o>[</span>insert<span class=w> </span>from<span class=w> </span>rKSHV.219-derived<span class=w> </span>BAC<span class=w> </span>clone<span class=w> </span>of<span class=w> </span>the<span class=w> </span>full-length<span class=w> </span>KSHV<span class=w> </span>genome<span class=o>]</span>

VIRUSES:
<span class=w>  </span>*<span class=w> </span>NC_007605.1<span class=w>   </span><span class=o>[</span>Human<span class=w> </span>gammaherpesvirus<span class=w> </span><span class=m>4</span><span class=w> </span><span class=o>(</span>Epstein-Barr<span class=w> </span>virus<span class=o>)]</span>
<span class=w>  </span>*<span class=w> </span>NC_006273.2<span class=w>   </span><span class=o>[</span>Human<span class=w> </span>betaherpesvirus<span class=w> </span><span class=m>5</span><span class=w> </span><span class=o>(</span>Cytomegalovirus<span class=w> </span><span class=o>)]</span>
<span class=w>  </span>*<span class=w> </span>NC_001664.4<span class=w>   </span><span class=o>[</span>Human<span class=w> </span>betaherpesvirus<span class=w> </span>6A<span class=w> </span><span class=o>(</span>HHV-6A<span class=o>)]</span>
<span class=w>  </span>*<span class=w> </span>NC_000898.1<span class=w>   </span><span class=o>[</span>Human<span class=w> </span>betaherpesvirus<span class=w> </span>6B<span class=w> </span><span class=o>(</span>HHV-6B<span class=o>)]</span>
<span class=w>  </span>*<span class=w> </span>NC_001716.2<span class=w>   </span><span class=o>[</span>Human<span class=w> </span>betaherpesvirus<span class=w> </span><span class=m>7</span><span class=w> </span><span class=o>(</span>HHV-7<span class=o>)]</span>
<span class=w>  </span>*<span class=w> </span>NC_009333.1<span class=w>   </span><span class=o>[</span>Human<span class=w> </span>gammaherpesvirus<span class=w> </span><span class=m>8</span><span class=w> </span><span class=o>(</span>KSHV<span class=o>)]</span>
<span class=w>  </span>*<span class=w> </span>NC_045512.2<span class=w>   </span><span class=o>[</span>Severe<span class=w> </span>acute<span class=w> </span>respiratory<span class=w> </span>syndrome<span class=o>(</span>SARS<span class=o>)</span>-related<span class=w> </span>coronavirus<span class=o>]</span>
<span class=w>  </span>*<span class=w> </span>MN485971.1<span class=w>    </span><span class=o>[</span>HIV<span class=w> </span>from<span class=w> </span>Belgium<span class=o>]</span>
<span class=w>  </span>*<span class=w> </span>NC_001806.2<span class=w>   </span><span class=o>[</span>Human<span class=w> </span>alphaherpesvirus<span class=w> </span><span class=m>1</span><span class=w> </span><span class=o>(</span>Herpes<span class=w> </span>simplex<span class=w> </span>virus<span class=w> </span><span class=nb>type</span><span class=w> </span><span class=m>1</span><span class=o>)](</span>strain<span class=w> </span><span class=m>17</span><span class=o>)</span><span class=w> </span><span class=o>(</span>HSV-1<span class=o>)]</span>
<span class=w>  </span>*<span class=w> </span>KT899744.1<span class=w>    </span><span class=o>[</span>HSV-1<span class=w> </span>strain<span class=w> </span>KOS<span class=o>]</span>
<span class=w>  </span>*<span class=w> </span>MH636806.1<span class=w>    </span><span class=o>[</span>MHV68<span class=w> </span><span class=o>(</span>Murine<span class=w> </span>herpesvirus<span class=w> </span><span class=m>68</span><span class=w> </span>strain<span class=w> </span>WUMS<span class=o>)]</span>

<span class=c1>##########################################################################################</span>

USAGE:
<span class=w>  </span>charlie<span class=w> </span>-w/--workdir<span class=o>=</span>&lt;WORKDIR&gt;<span class=w> </span>-m/--runmode<span class=o>=</span>&lt;RUNMODE&gt;

Required<span class=w> </span>Arguments:
<span class=m>1</span>.<span class=w>  </span>WORKDIR<span class=w>     </span>:<span class=w> </span><span class=o>[</span>Type:<span class=w> </span>String<span class=o>]</span>:<span class=w> </span>Absolute<span class=w> </span>or<span class=w> </span>relative<span class=w> </span>path<span class=w> </span>to<span class=w> </span>the<span class=w> </span>output<span class=w> </span>folder<span class=w> </span>with<span class=w> </span>write<span class=w> </span>permissions.

<span class=m>2</span>.<span class=w>  </span>RUNMODE<span class=w>     </span>:<span class=w> </span><span class=o>[</span>Type:<span class=w> </span>String<span class=o>]</span><span class=w> </span>Valid<span class=w> </span>options:
<span class=w>    </span>*<span class=w> </span>init<span class=w>      </span>:<span class=w> </span>initialize<span class=w> </span>workdir
<span class=w>    </span>*<span class=w> </span>dryrun<span class=w>    </span>:<span class=w> </span>dry<span class=w> </span>run<span class=w> </span>snakemake<span class=w> </span>to<span class=w> </span>generate<span class=w> </span>DAG
<span class=w>    </span>*<span class=w> </span>run<span class=w>       </span>:<span class=w> </span>run<span class=w> </span>with<span class=w> </span>slurm
<span class=w>    </span>*<span class=w> </span>runlocal<span class=w>  </span>:<span class=w> </span>run<span class=w> </span>without<span class=w> </span>submitting<span class=w> </span>to<span class=w> </span>sbatch
<span class=w>    </span>ADVANCED<span class=w> </span>RUNMODES<span class=w> </span><span class=o>(</span>use<span class=w> </span>with<span class=w> </span>caution!!<span class=o>)</span>
<span class=w>    </span>*<span class=w> </span>unlock<span class=w>    </span>:<span class=w> </span>unlock<span class=w> </span>WORKDIR<span class=w> </span><span class=k>if</span><span class=w> </span>locked<span class=w> </span>by<span class=w> </span>snakemake<span class=w> </span>NEVER<span class=w> </span>UNLOCK<span class=w> </span>WORKDIR<span class=w> </span>WHERE<span class=w> </span>PIPELINE<span class=w> </span>IS<span class=w> </span>CURRENTLY<span class=w> </span>RUNNING!
<span class=w>    </span>*<span class=w> </span>reconfig<span class=w>  </span>:<span class=w> </span>recreate<span class=w> </span>config<span class=w> </span>file<span class=w> </span><span class=k>in</span><span class=w> </span>WORKDIR<span class=w> </span><span class=o>(</span>debugging<span class=w> </span>option<span class=o>)</span><span class=w> </span>EDITS<span class=w> </span>TO<span class=w> </span>config.yaml<span class=w> </span>WILL<span class=w> </span>BE<span class=w> </span>LOST!
<span class=w>    </span>*<span class=w> </span>reset<span class=w>     </span>:<span class=w> </span>DELETE<span class=w> </span>workdir<span class=w> </span>dir<span class=w> </span>and<span class=w> </span>re-init<span class=w> </span>it<span class=w> </span><span class=o>(</span>debugging<span class=w> </span>option<span class=o>)</span><span class=w> </span>EDITS<span class=w> </span>TO<span class=w> </span>ALL<span class=w> </span>FILES<span class=w> </span>IN<span class=w> </span>WORKDIR<span class=w> </span>WILL<span class=w> </span>BE<span class=w> </span>LOST!
<span class=w>    </span>*<span class=w> </span>printbinds:<span class=w> </span>print<span class=w> </span>singularity<span class=w> </span>binds<span class=w> </span><span class=o>(</span>paths<span class=o>)</span>
<span class=w>    </span>*<span class=w> </span><span class=nb>local</span><span class=w>     </span>:<span class=w> </span>same<span class=w> </span>as<span class=w> </span>runlocal

Optional<span class=w> </span>Arguments:

--singcache<span class=p>|</span>-c<span class=w>  </span>:<span class=w> </span>singularity<span class=w> </span>cache<span class=w> </span>directory.<span class=w> </span>Default<span class=w> </span>is<span class=w> </span><span class=sb>`</span>/data/<span class=si>${</span><span class=nv>USER</span><span class=si>}</span>/.singularity<span class=sb>`</span><span class=w> </span><span class=k>if</span><span class=w> </span>available,<span class=w> </span>or<span class=w> </span>falls<span class=w> </span>back<span class=w> </span>to<span class=w> </span><span class=sb>`</span><span class=si>${</span><span class=nv>WORKDIR</span><span class=si>}</span>/.singularity<span class=sb>`</span>.<span class=w> </span>Use<span class=w> </span>this<span class=w> </span>flag<span class=w> </span>to<span class=w> </span>specify<span class=w> </span>a<span class=w> </span>different<span class=w> </span>singularity<span class=w> </span>cache<span class=w> </span>directory.
--host<span class=p>|</span>-g<span class=w>       </span>:<span class=w> </span>supply<span class=w> </span>host<span class=w> </span>at<span class=w> </span><span class=nb>command</span><span class=w> </span>line.<span class=w> </span>hg38<span class=w> </span>or<span class=w> </span>mm39.<span class=w>                                            </span><span class=o>(</span>--runmode<span class=o>=</span>init<span class=w> </span>only<span class=o>)</span>
--additives<span class=p>|</span>-a<span class=w>  </span>:<span class=w> </span>supply<span class=w> </span>comma-separated<span class=w> </span>list<span class=w> </span>of<span class=w> </span>additives<span class=w> </span>at<span class=w> </span><span class=nb>command</span><span class=w> </span>line.<span class=w> </span>ERCC<span class=w> </span>or<span class=w> </span>BAC16Insert<span class=w> </span>or<span class=w> </span>both<span class=w> </span><span class=o>(</span>--runmode<span class=o>=</span>init<span class=w> </span>only<span class=o>)</span>
--viruses<span class=p>|</span>-v<span class=w>    </span>:<span class=w> </span>supply<span class=w> </span>comma-separated<span class=w> </span>list<span class=w> </span>of<span class=w> </span>viruses<span class=w> </span>at<span class=w> </span><span class=nb>command</span><span class=w> </span>line<span class=w>                                </span><span class=o>(</span>--runmode<span class=o>=</span>init<span class=w> </span>only<span class=o>)</span>
--manifest<span class=p>|</span>-s<span class=w>   </span>:<span class=w> </span>absolute<span class=w> </span>path<span class=w> </span>to<span class=w> </span>samples.tsv.<span class=w> </span>This<span class=w> </span>will<span class=w> </span>be<span class=w> </span>copied<span class=w> </span>to<span class=w> </span>output<span class=w> </span>folder<span class=w>                    </span><span class=o>(</span>--runmode<span class=o>=</span>init<span class=w> </span>only<span class=o>)</span>
--changegrp<span class=p>|</span>-z<span class=w>  </span>:<span class=w> </span>change<span class=w> </span>group<span class=w> </span>to<span class=w> </span><span class=s2>&quot;Ziegelbauer_lab&quot;</span><span class=w> </span>before<span class=w> </span>running<span class=w> </span>anything.<span class=w> </span>Biowulf-only.<span class=w> </span>Useful<span class=w> </span><span class=k>for</span><span class=w> </span>correctly<span class=w> </span>setting<span class=w> </span>permissions.
--help<span class=p>|</span>-h<span class=w>       </span>:<span class=w> </span>print<span class=w> </span>this<span class=w> </span><span class=nb>help</span>


Example<span class=w> </span>commands:
<span class=w>  </span>charlie<span class=w> </span>-w<span class=o>=</span>/my/output/folder<span class=w> </span>-m<span class=o>=</span>init
<span class=w>  </span>charlie<span class=w> </span>-w<span class=o>=</span>/my/output/folder<span class=w> </span>-m<span class=o>=</span>dryrun
<span class=w>  </span>charlie<span class=w> </span>-w<span class=o>=</span>/my/output/folder<span class=w> </span>-m<span class=o>=</span>run

<span class=c1>##########################################################################################</span>

VersionInfo:
<span class=w>  </span>python<span class=w>          </span>:<span class=w> </span><span class=m>3</span>
<span class=w>  </span>snakemake<span class=w>       </span>:<span class=w> </span><span class=m>7</span>
<span class=w>  </span>pipeline_home<span class=w>   </span>:<span class=w> </span>/gpfs/gsfs10/users/CCBR_Pipeliner/Pipelines/CHARLIE/.v0.11.1
<span class=w>  </span>git<span class=w> </span>commit/tag<span class=w>  </span>:<span class=w> </span>613fb617f1ed426fb8900f98e599ca0497a67cc0<span class=w>    </span>v0.11.0-49-g613fb61

<span class=c1>##########################################################################################</span>
</code></pre></div> <h3 id=5-license>5. License<a class=headerlink href=#5-license title="Permanent link">&para;</a></h3> <p>MIT License</p> <p>Copyright &copy; 2021 Vishal Koparde</p> <p>Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:</p> <p>The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.</p> <p>THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.</p> <h3 id=6-testing>6. Testing<a class=headerlink href=#6-testing title="Permanent link">&para;</a></h3> <h4 id=init>Init<a class=headerlink href=#init title="Permanent link">&para;</a></h4> <p>Run init mode:</p> <div class=highlight><pre><span></span><code>bash<span class=w> </span>&lt;path<span class=w> </span>to<span class=w> </span>charlie&gt;<span class=w> </span>-w<span class=o>=</span>&lt;path<span class=w> </span>to<span class=w> </span>output<span class=w> </span>dir&gt;<span class=w> </span>-m<span class=o>=</span>init
</code></pre></div> <p>This will create the folder provided by <code>-w=</code>. The user should have write permission to this folder.</p> <h4 id=dry-run>Dry-run<a class=headerlink href=#dry-run title="Permanent link">&para;</a></h4> <p>Test data (1 paired-end subsample and 1 single-end subsample) have been including under the <code>/data/CCBR_Pipeliner/testdata/circRNA/human</code> folder. After running in <code>-m=init</code>, <code>samples.tsv</code> should be edited to point the copies of the above mentioned samples with the column headers:</p> <ul> <li>sampleName</li> <li>path_to_R1_fastq</li> <li>path_to_R2_fastq</li> </ul> <p>Column <code>path_to_R2_fastq</code> will be blank in case of single-end samples.</p> <p>After editing <code>samples.tsv</code>, dry run should be run:</p> <div class=highlight><pre><span></span><code>bash<span class=w> </span>&lt;path<span class=w> </span>to<span class=w> </span>charlie&gt;<span class=w> </span>-w<span class=o>=</span>&lt;path<span class=w> </span>to<span class=w> </span>output<span class=w> </span>dir&gt;<span class=w> </span>-m<span class=o>=</span>dryrun
</code></pre></div> <p>This will create the reference fasta and gtf file based on the selections made in the <code>config.yaml</code>.</p> <h4 id=run>Run<a class=headerlink href=#run title="Permanent link">&para;</a></h4> <p>If <code>-m=dryrun</code> was successful, then simply do <code>-m=run</code>. The output will look something like this</p> <div class=highlight><pre><span></span><code>... ... skipping ~1000 lines
...
...
Job stats:
job                                              count    min threads    max threads
---------------------------------------------  -------  ----------
all                                                  1              1              1
annotate_clear_output                                2              1              1
circExplorer                                         2              2              2
circExplorer_bwa                                     2              2              2
circrnafinder                                        2              1              1
ciri                                                 2             56             56
clear                                                2              2              2
create_bowtie2_index                                 1              1              1
create_bwa_index                                     1              1              1
create_circExplorer_BSJ_bam                          2              4              4
create_circExplorer_linear_spliced_bams              2             56             56
create_circExplorer_merged_found_counts_table        2              1              1
create_hq_bams                                       2              1              1
create_index                                         1             56             56
create_master_counts_file                            1              1              1
cutadapt                                             2             56             56
dcc                                                  2              4              4
dcc_create_samplesheets                              2              1              1
estimate_duplication                                 2              1              1
fastqc                                               2              4              4
find_circ                                            2             56             56
find_circ_align                                      2             56             56
merge_SJ_tabs                                        1              2              2
merge_alignment_stats                                1              1              1
merge_genecounts                                     1              1              1
merge_per_sample                                     2              1              1
star1p                                               2             56             56
star2p                                               2             56             56
star_circrnafinder                                   2             56             56
total                                               52              1             56

Reasons:
    (check individual jobs above for details)
    input files updated by another job:
        alignment_stats, all, annotate_clear_output, circExplorer, circExplorer_bwa, circrnafinder, ciri, clear, create_circExplorer_BSJ_bam, create_circExplorer_linear_spliced_bams, create_circExplorer_merged_found_counts_table, create_hq_bams, create_master_counts_file, dcc, dcc_create_samplesheets, estimate_duplication, fastqc, find_circ, find_circ_align, merge_SJ_tabs, merge_alignment_stats, merge_genecounts, merge_per_sample, star1p, star2p, star_circrnafinder
    missing output files:
        alignment_stats, annotate_clear_output, circExplorer, circExplorer_bwa, circrnafinder, ciri, clear, create_bowtie2_index, create_bwa_index, create_circExplorer_BSJ_bam, create_circExplorer_linear_spliced_bams, create_circExplorer_merged_found_counts_table, create_hq_bams, create_index, create_master_counts_file, cutadapt, dcc, dcc_create_samplesheets, estimate_duplication, fastqc, find_circ, find_circ_align, merge_SJ_tabs, merge_alignment_stats, merge_genecounts, merge_per_sample, star1p, star2p, star_circrnafinder

This was a dry-run (flag -n). The order of jobs does not reflect the order of execution.
Running...
14743440
</code></pre></div> <h5 id=61-test-data>6.1 Test Data<a class=headerlink href=#61-test-data title="Permanent link">&para;</a></h5> <p>The <code>/data/CCBR_Pipeliner/testdata/circRNA/human</code> folder in the repo has test dataset:</p> <div class=highlight><pre><span></span><code>tree<span class=w> </span>/data/CCBR_Pipeliner/testdata/circRNA/human
/data/CCBR_Pipeliner/testdata/circRNA/human
├──<span class=w> </span>GI1_N_ss.R1.fastq.gz
├──<span class=w> </span>GI1_N_ss.R2.fastq.gz
├──<span class=w> </span>GI1_T_ss.R1.fastq.gz
└──<span class=w> </span>samples.tsv
</code></pre></div> <p><code>GI1_N</code> is a PE sample while <code>GI1_T</code> is a SE sample.</p> <h5 id=62-expected-output>6.2 Expected Output<a class=headerlink href=#62-expected-output title="Permanent link">&para;</a></h5> <p>Expected output from the sample data is stored under <code>.tests/expected_output</code>.</p> <p>More details about running test data can be found <a href=https://ccbr.github.io/CHARLIE/tutorial>here</a>.</p> <blockquote> <p>DISCLAIMER:</p> <p>CHARLIE is built to be run only on <a href=https://hpc.nih.gov>BIOWULF</a>. A newer HPC-agnostic version of CHARLIE is planned for 2024.</p> </blockquote> <!-- the contents are automatically inserted here. do not edit this file, instead edit the file specified above to make changes. --> </div> </div><script async src="https://www.googletagmanager.com/gtag/js?id=G-D3SL9V30KL"></script> <script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());
  gtag('config', 'G-D3SL9V30KL'); // Replace with your tracking ID
</script> </div> </div> </section> </div> <div class=rst-versions role=note aria-label=Versions> <span class=rst-current-version data-toggle=rst-current-version> <span> <a href=https://github.com/CCBR/CHARLIE class="fa fa-code-fork" style="color: #fcfcfc"> CCBR/CHARLIE</a> </span> <span><a href=flowchart/ style="color: #fcfcfc">Next &raquo;</a></span> </span> </div> <script src=js/jquery-3.6.0.min.js></script> <script>var base_url = ".";</script> <script src=js/theme_extra.js></script> <script src=js/theme.js></script> <script src=js/custom.js></script> <script src=search/main.js></script> <script src=js/version-select.js></script> <script>
        jQuery(function () {
            SphinxRtdTheme.Navigation.enable(true);
        });
    </script> </body> </html> <!--
MkDocs version : 1.6.1
Build Date UTC : 2025-04-10 16:41:54.100910+00:00
--> 