<!doctype html><html lang=en class=no-js> <head><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><meta name=author content="Vishal Koparde, Ph.D."><link href=flowchart/ rel=next><link rel=icon href=assets/images/favicon.png><meta name=generator content="mkdocs-1.5.3, mkdocs-material-9.1.6"><title>CHARLIE (Circrnas in Host And viRuses anaLysis pIpEline)</title><link rel=stylesheet href=assets/stylesheets/main.ded33207.min.css><link rel=stylesheet href=assets/stylesheets/palette.a0c5b2b5.min.css><link rel=preconnect href=https://fonts.gstatic.com crossorigin><link rel=stylesheet href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback"><style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style><script>__md_scope=new URL(".",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script></head> <body dir=ltr data-md-color-scheme=default data-md-color-primary=indigo data-md-color-accent=indigo> <script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script> <input class=md-toggle data-md-toggle=drawer type=checkbox id=__drawer autocomplete=off> <input class=md-toggle data-md-toggle=search type=checkbox id=__search autocomplete=off> <label class=md-overlay for=__drawer></label> <div data-md-component=skip> <a href=#charlie class=md-skip> Skip to content </a> </div> <div data-md-component=announce> </div> <div data-md-color-scheme=default data-md-component=outdated hidden> </div> <header class=md-header data-md-component=header> <nav class="md-header__inner md-grid" aria-label=Header> <a href=. title="CHARLIE (Circrnas in Host And viRuses anaLysis pIpEline)" class="md-header__button md-logo" aria-label="CHARLIE (Circrnas in Host And viRuses anaLysis pIpEline)" data-md-component=logo> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg> </a> <label class="md-header__button md-icon" for=__drawer> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg> </label> <div class=md-header__title data-md-component=header-title> <div class=md-header__ellipsis> <div class=md-header__topic> <span class=md-ellipsis> CHARLIE (Circrnas in Host And viRuses anaLysis pIpEline) </span> </div> <div class=md-header__topic data-md-component=header-topic> <span class=md-ellipsis> Home </span> </div> </div> </div> <form class=md-header__option data-md-component=palette> <input class=md-option data-md-color-media data-md-color-scheme=default data-md-color-primary=indigo data-md-color-accent=indigo aria-label="Switch to dark mode" type=radio name=__palette id=__palette_1> <label class="md-header__button md-icon" title="Switch to dark mode" for=__palette_2 hidden> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M17 6H7c-3.31 0-6 2.69-6 6s2.69 6 6 6h10c3.31 0 6-2.69 6-6s-2.69-6-6-6zm0 10H7c-2.21 0-4-1.79-4-4s1.79-4 4-4h10c2.21 0 4 1.79 4 4s-1.79 4-4 4zM7 9c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"/></svg> </label> <input class=md-option data-md-color-media data-md-color-scheme=slate data-md-color-primary=red data-md-color-accent=red aria-label="Switch to light mode" type=radio name=__palette id=__palette_2> <label class="md-header__button md-icon" title="Switch to light mode" for=__palette_1 hidden> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M17 7H7a5 5 0 0 0-5 5 5 5 0 0 0 5 5h10a5 5 0 0 0 5-5 5 5 0 0 0-5-5m0 8a3 3 0 0 1-3-3 3 3 0 0 1 3-3 3 3 0 0 1 3 3 3 3 0 0 1-3 3Z"/></svg> </label> </form> <label class="md-header__button md-icon" for=__search> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg> </label> <div class=md-search data-md-component=search role=dialog> <label class=md-search__overlay for=__search></label> <div class=md-search__inner role=search> <form class=md-search__form name=search> <input type=text class=md-search__input name=query aria-label=Search placeholder=Search autocapitalize=off autocorrect=off autocomplete=off spellcheck=false data-md-component=search-query required> <label class="md-search__icon md-icon" for=__search> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg> </label> <nav class=md-search__options aria-label=Search> <button type=reset class="md-search__icon md-icon" title=Clear aria-label=Clear tabindex=-1> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg> </button> </nav> </form> <div class=md-search__output> <div class=md-search__scrollwrap data-md-scrollfix> <div class=md-search-result data-md-component=search-result> <div class=md-search-result__meta> Initializing search </div> <ol class=md-search-result__list role=presentation></ol> </div> </div> </div> </div> </div> <div class=md-header__source> <a href=https://github.com/CCBR/CHARLIE title="Go to repository" class=md-source data-md-component=source> <div class="md-source__icon md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 448 512"><!-- Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg> </div> <div class=md-source__repository> CCBR/CHARLIE </div> </a> </div> </nav> </header> <div class=md-container data-md-component=container> <nav class=md-tabs aria-label=Tabs data-md-component=tabs> <div class=md-grid> <ul class=md-tabs__list> <li class=md-tabs__item> <a href=. class="md-tabs__link md-tabs__link--active"> Home </a> </li> <li class=md-tabs__item> <a href=flowchart/ class=md-tabs__link> Workflow </a> </li> <li class=md-tabs__item> <a href=tutorial/ class=md-tabs__link> Tutorial </a> </li> <li class=md-tabs__item> <a href=references/ class=md-tabs__link> References </a> </li> <li class=md-tabs__item> <a href=CHANGELOG/ class=md-tabs__link> Changelog </a> </li> <li class=md-tabs__item> <a href=contributing/ class=md-tabs__link> How to contribute </a> </li> </ul> </div> </nav> <main class=md-main data-md-component=main> <div class="md-main__inner md-grid"> <div class="md-sidebar md-sidebar--primary" data-md-component=sidebar data-md-type=navigation> <div class=md-sidebar__scrollwrap> <div class=md-sidebar__inner> <nav class="md-nav md-nav--primary md-nav--lifted md-nav--integrated" aria-label=Navigation data-md-level=0> <label class=md-nav__title for=__drawer> <a href=. title="CHARLIE (Circrnas in Host And viRuses anaLysis pIpEline)" class="md-nav__button md-logo" aria-label="CHARLIE (Circrnas in Host And viRuses anaLysis pIpEline)" data-md-component=logo> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg> </a> CHARLIE (Circrnas in Host And viRuses anaLysis pIpEline) </label> <div class=md-nav__source> <a href=https://github.com/CCBR/CHARLIE title="Go to repository" class=md-source data-md-component=source> <div class="md-source__icon md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 448 512"><!-- Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg> </div> <div class=md-source__repository> CCBR/CHARLIE </div> </a> </div> <ul class=md-nav__list data-md-scrollfix> <li class="md-nav__item md-nav__item--active"> <input class="md-nav__toggle md-toggle" type=checkbox id=__toc> <label class="md-nav__link md-nav__link--active" for=__toc> Home <span class="md-nav__icon md-icon"></span> </label> <a href=. class="md-nav__link md-nav__link--active"> Home </a> <nav class="md-nav md-nav--secondary" aria-label="Table of contents"> <label class=md-nav__title for=__toc> <span class="md-nav__icon md-icon"></span> Table of contents </label> <ul class=md-nav__list data-md-component=toc data-md-scrollfix> <li class=md-nav__item> <a href=#table-of-contents class=md-nav__link> Table of Contents </a> </li> <li class=md-nav__item> <a href=#1-introduction class=md-nav__link> 1. Introduction </a> </li> <li class=md-nav__item> <a href=#2-flowchart class=md-nav__link> 2. Flowchart </a> </li> <li class=md-nav__item> <a href=#3-software-dependencies class=md-nav__link> 3. Software Dependencies </a> </li> <li class=md-nav__item> <a href=#4-usage class=md-nav__link> 4. Usage </a> </li> <li class=md-nav__item> <a href=#5-license class=md-nav__link> 5. License </a> </li> <li class=md-nav__item> <a href=#6-testing class=md-nav__link> 6. Testing </a> <nav class=md-nav aria-label="6. Testing"> <ul class=md-nav__list> <li class=md-nav__item> <a href=#init class=md-nav__link> Init </a> </li> <li class=md-nav__item> <a href=#dry-run class=md-nav__link> Dry-run </a> </li> <li class=md-nav__item> <a href=#run class=md-nav__link> Run </a> <nav class=md-nav aria-label=Run> <ul class=md-nav__list> <li class=md-nav__item> <a href=#61-test-data class=md-nav__link> 6.1 Test Data </a> </li> <li class=md-nav__item> <a href=#62-expected-output class=md-nav__link> 6.2 Expected Output </a> </li> </ul> </nav> </li> </ul> </nav> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=flowchart/ class=md-nav__link> Workflow </a> </li> <li class=md-nav__item> <a href=tutorial/ class=md-nav__link> Tutorial </a> </li> <li class=md-nav__item> <a href=references/ class=md-nav__link> References </a> </li> <li class=md-nav__item> <a href=CHANGELOG/ class=md-nav__link> Changelog </a> </li> <li class=md-nav__item> <a href=contributing/ class=md-nav__link> How to contribute </a> </li> </ul> </nav> </div> </div> </div> <div class=md-content data-md-component=content> <article class="md-content__inner md-typeset"> <h1 id=charlie>CHARLIE<a class=headerlink href=#charlie title="Permanent link">&para;</a></h1> <p><img alt=img src="https://img.shields.io/github/issues/CCBR/CHARLIE?style=for-the-badge"><img alt=img src="https://img.shields.io/github/forks/CCBR/CHARLIE?style=for-the-badge"><img alt=img src="https://img.shields.io/github/stars/CCBR/CHARLIE?style=for-the-badge"><img alt=img src="https://img.shields.io/github/license/CCBR/CHARLIE?style=for-the-badge"></p> <h3 id=table-of-contents>Table of Contents<a class=headerlink href=#table-of-contents title="Permanent link">&para;</a></h3> <ul> <li><a href=#charlie>CHARLIE - <strong>C</strong>ircrnas in <strong>H</strong>ost <strong>A</strong>nd vi<strong>R</strong>uses ana<strong>L</strong>ysis p<strong>I</strong>p<strong>E</strong>line</a></li> <li><a href=#table-of-contents>Table of Contents</a></li> <li><a href=#1-introduction>1. Introduction</a></li> <li><a href=#2-flowchart>2. Flowchart</a></li> <li><a href=#3-software-dependencies>3. Software Dependencies</a></li> <li><a href=#4-usage>4. Usage</a></li> <li><a href=#5-license>5. License</a></li> <li><a href=#6-testing>6. Testing</a><ul> <li><a href=#61-test-data>6.1 Test data</a></li> <li><a href=#62-expected-output>6.2 Expected output</a></li> </ul> </li> </ul> <h3 id=1-introduction>1. Introduction<a class=headerlink href=#1-introduction title="Permanent link">&para;</a></h3> <p><strong>C</strong>ircrnas in <strong>H</strong>ost <strong>A</strong>nd vi<strong>R</strong>uses ana<strong>L</strong>ysis p<strong>I</strong>p<strong>E</strong>line</p> <p>Things to know about CHARLIE:</p> <ul> <li>Snakemake workflow to detect, annotate and quantify (DAQ) host and viral circular RNAs.</li> <li>Primirarily developed to run on <a href=https://hpc.nih.gov/ >BIOWULF</a></li> <li>Reach out to <a href=mailto:vishal.koparde@nihgov>Vishal Koparde</a> for questions/comments/requests.</li> </ul> <p>This circularRNA detection pipeline uses CIRCExplorer2, CIRI2 and many other tools in parallel to detect, quantify and annotate circRNAs. Here is a list of tools that can be run using CHARLIE:</p> <table> <thead> <tr> <th>circRNA Detection Tool</th> <th>Aligner(s)</th> <th>Run by default</th> </tr> </thead> <tbody> <tr> <td><a href=https://github.com/YangLab/CIRCexplorer2>CIRCExplorer2</a></td> <td>STAR<sup>1</sup></td> <td>Yes</td> </tr> <tr> <td><a href=https://sourceforge.net/projects/ciri/files/CIRI2/ >CIRI2</a></td> <td>BWA<sup>1</sup></td> <td>Yes</td> </tr> <tr> <td><a href=https://github.com/YangLab/CIRCexplorer2>CIRCExplorer2</a></td> <td>BWA<sup>1</sup></td> <td>Yes</td> </tr> <tr> <td><a href=https://github.com/YangLab/CLEAR>CLEAR</a></td> <td>STAR<sup>1</sup></td> <td>Yes</td> </tr> <tr> <td><a href=https://github.com/dieterich-lab/DCC>DCC</a></td> <td>STAR<sup>2</sup></td> <td>Yes</td> </tr> <tr> <td><a href=https://github.com/bioxfu/circRNAFinder>circRNAFinder</a></td> <td>STAR<sup>3</sup></td> <td>Yes</td> </tr> <tr> <td><a href=https://github.com/marvin-jens/find_circ>find_circ</a></td> <td>Bowtie2</td> <td>Yes</td> </tr> <tr> <td><a href=https://github.com/merckey/MapSplice2>MapSplice</a></td> <td>BWA<sup>2</sup></td> <td>No</td> </tr> <tr> <td><a href=https://github.com/TreesLab/NCLscan>NCLScan</a></td> <td>NovoAlign</td> <td>No</td> </tr> </tbody> </table> <blockquote> <p>Note: STAR<sup>1</sup>, STAR<sup>2</sup>, STAR<sup>3</sup> denote 3 different sets of alignment parameters, etc.</p> <p>Note: BWA<sup>1</sup>, BWA<sup>2</sup> denote 2 different alignment parameters, etc.</p> </blockquote> <h3 id=2-flowchart>2. Flowchart<a class=headerlink href=#2-flowchart title="Permanent link">&para;</a></h3> <p><img alt src=docs/images/CHARLIE_v0.8.x.png></p> <p>For complete documentation with tutorial go <a href=https://CCBR.github.io/CHARLIE/ >here</a>.</p> <blockquote> <p>DISCLAIMER: New circRNA tools have been added CHARLIE and the documentation is currently out of date!</p> </blockquote> <h3 id=3-software-dependencies>3. Software Dependencies<a class=headerlink href=#3-software-dependencies title="Permanent link">&para;</a></h3> <p>The following version of various bioinformatics tools are using within CHARLIE:</p> <table> <thead> <tr> <th>tool</th> <th>version</th> </tr> </thead> <tbody> <tr> <td>blat</td> <td>3.5</td> </tr> <tr> <td>bedtools</td> <td>2.30.0</td> </tr> <tr> <td>bowtie</td> <td>2-2.5.1</td> </tr> <tr> <td>bowtie</td> <td>1.3.1</td> </tr> <tr> <td>bwa</td> <td>0.7.17</td> </tr> <tr> <td>circexplorer2</td> <td>2.3.8</td> </tr> <tr> <td>cufflinks</td> <td>2.2.1</td> </tr> <tr> <td>cutadapt</td> <td>4.4</td> </tr> <tr> <td>fastqc</td> <td>0.11.9</td> </tr> <tr> <td>hisat</td> <td>2.2.2.1</td> </tr> <tr> <td>java</td> <td>18.0.1.1</td> </tr> <tr> <td>multiqc</td> <td>1.9</td> </tr> <tr> <td>parallel</td> <td>20231122</td> </tr> <tr> <td>perl</td> <td>5.34</td> </tr> <tr> <td>picard</td> <td>2.27.3</td> </tr> <tr> <td>python</td> <td>2.7</td> </tr> <tr> <td>python</td> <td>3.8</td> </tr> <tr> <td>sambamba</td> <td>0.8.2</td> </tr> <tr> <td>samtools</td> <td>1.16.1</td> </tr> <tr> <td>STAR</td> <td>2.7.6a</td> </tr> <tr> <td>stringtie</td> <td>2.2.1</td> </tr> <tr> <td>ucsc</td> <td>450</td> </tr> <tr> <td>R</td> <td>4.0.5</td> </tr> <tr> <td>novocraft</td> <td>4.03.05</td> </tr> </tbody> </table> <h3 id=4-usage>4. Usage<a class=headerlink href=#4-usage title="Permanent link">&para;</a></h3> <div class=highlight><pre><span></span><code><span class=w> </span>%<span class=w> </span>./charlie


<span class=c1>##########################################################################################</span>

Welcome<span class=w> </span>to
<span class=w> </span>_______<span class=w>  </span>__<span class=w>   </span>__<span class=w>  </span>_______<span class=w>  </span>______<span class=w>    </span>___<span class=w>      </span>___<span class=w>   </span>_______
<span class=p>|</span><span class=w>       </span><span class=o>||</span><span class=w>  </span><span class=p>|</span><span class=w> </span><span class=p>|</span><span class=w>  </span><span class=o>||</span><span class=w>   </span>_<span class=w>   </span><span class=o>||</span><span class=w>    </span>_<span class=w> </span><span class=p>|</span><span class=w>  </span><span class=p>|</span><span class=w>   </span><span class=p>|</span><span class=w>    </span><span class=p>|</span><span class=w>   </span><span class=p>|</span><span class=w> </span><span class=p>|</span><span class=w>       </span><span class=p>|</span>
<span class=p>|</span><span class=w>       </span><span class=o>||</span><span class=w>  </span><span class=p>|</span>_<span class=p>|</span><span class=w>  </span><span class=o>||</span><span class=w>  </span><span class=p>|</span>_<span class=p>|</span><span class=w>  </span><span class=o>||</span><span class=w>   </span><span class=p>|</span><span class=w> </span><span class=o>||</span><span class=w>  </span><span class=p>|</span><span class=w>   </span><span class=p>|</span><span class=w>    </span><span class=p>|</span><span class=w>   </span><span class=p>|</span><span class=w> </span><span class=p>|</span><span class=w>    </span>___<span class=p>|</span>
<span class=p>|</span><span class=w>       </span><span class=o>||</span><span class=w>       </span><span class=o>||</span><span class=w>       </span><span class=o>||</span><span class=w>   </span><span class=p>|</span>_<span class=o>||</span>_<span class=w> </span><span class=p>|</span><span class=w>   </span><span class=p>|</span><span class=w>    </span><span class=p>|</span><span class=w>   </span><span class=p>|</span><span class=w> </span><span class=p>|</span><span class=w>   </span><span class=p>|</span>___
<span class=p>|</span><span class=w>      </span>_<span class=o>||</span><span class=w>       </span><span class=o>||</span><span class=w>       </span><span class=o>||</span><span class=w>    </span>__<span class=w>  </span><span class=o>||</span><span class=w>   </span><span class=p>|</span>___<span class=w> </span><span class=p>|</span><span class=w>   </span><span class=p>|</span><span class=w> </span><span class=p>|</span><span class=w>    </span>___<span class=p>|</span>
<span class=p>|</span><span class=w>     </span><span class=p>|</span>_<span class=w> </span><span class=p>|</span><span class=w>   </span>_<span class=w>   </span><span class=o>||</span><span class=w>   </span>_<span class=w>   </span><span class=o>||</span><span class=w>   </span><span class=p>|</span><span class=w>  </span><span class=p>|</span><span class=w> </span><span class=o>||</span><span class=w>       </span><span class=o>||</span><span class=w>   </span><span class=p>|</span><span class=w> </span><span class=p>|</span><span class=w>   </span><span class=p>|</span>___
<span class=p>|</span>_______<span class=o>||</span>__<span class=p>|</span><span class=w> </span><span class=p>|</span>__<span class=o>||</span>__<span class=p>|</span><span class=w> </span><span class=p>|</span>__<span class=o>||</span>___<span class=p>|</span><span class=w>  </span><span class=p>|</span>_<span class=o>||</span>_______<span class=o>||</span>___<span class=p>|</span><span class=w> </span><span class=p>|</span>_______<span class=p>|</span>

C_ircrnas<span class=w> </span><span class=k>in</span><span class=w> </span>H_ost<span class=w> </span>A_nd<span class=w> </span>vi_R_uses<span class=w> </span>ana_L_ysis<span class=w> </span>p_I_p_E_line

<span class=c1>##########################################################################################</span>

This<span class=w> </span>pipeline<span class=w> </span>was<span class=w> </span>built<span class=w> </span>by<span class=w> </span>CCBR<span class=w> </span><span class=o>(</span>https://bioinformatics.ccr.cancer.gov/ccbr<span class=o>)</span>
Please<span class=w> </span>contact<span class=w> </span>Vishal<span class=w> </span>Koparde<span class=w> </span><span class=k>for</span><span class=w> </span>comments/questions<span class=w> </span><span class=o>(</span>vishal.koparde@nih.gov<span class=o>)</span>

<span class=c1>##########################################################################################</span>

CHARLIE<span class=w> </span>can<span class=w> </span>be<span class=w> </span>used<span class=w> </span>to<span class=w> </span>DAQ<span class=o>(</span>Detect/Annotate/Quantify<span class=o>)</span><span class=w> </span>circRNAs<span class=w> </span><span class=k>in</span><span class=w> </span>hosts<span class=w> </span>and<span class=w> </span>viruses.

Here<span class=w> </span>is<span class=w> </span>the<span class=w> </span>list<span class=w> </span>of<span class=w> </span>hosts<span class=w> </span>and<span class=w> </span>viruses<span class=w> </span>that<span class=w> </span>are<span class=w> </span>currently<span class=w> </span>supported:

HOSTS:
<span class=w>  </span>*<span class=w> </span>hg38<span class=w>          </span><span class=o>[</span>Human<span class=o>]</span>
<span class=w>  </span>*<span class=w> </span>mm39<span class=w>          </span><span class=o>[</span>Mouse<span class=o>]</span>

ADDITIVES:
<span class=w>  </span>*<span class=w> </span>ERCC<span class=w>          </span><span class=o>[</span>External<span class=w> </span>RNA<span class=w> </span>Control<span class=w> </span>Consortium<span class=w> </span>sequences<span class=o>]</span>
<span class=w>  </span>*<span class=w> </span>BAC16Insert<span class=w>   </span><span class=o>[</span>insert<span class=w> </span>from<span class=w> </span>rKSHV.219-derived<span class=w> </span>BAC<span class=w> </span>clone<span class=w> </span>of<span class=w> </span>the<span class=w> </span>full-length<span class=w> </span>KSHV<span class=w> </span>genome<span class=o>]</span>

VIRUSES:
<span class=w>  </span>*<span class=w> </span>NC_007605.1<span class=w>   </span><span class=o>[</span>Human<span class=w> </span>gammaherpesvirus<span class=w> </span><span class=m>4</span><span class=w> </span><span class=o>(</span>Epstein-Barr<span class=w> </span>virus<span class=o>)]</span>
<span class=w>  </span>*<span class=w> </span>NC_006273.2<span class=w>   </span><span class=o>[</span>Human<span class=w> </span>betaherpesvirus<span class=w> </span><span class=m>5</span><span class=w> </span><span class=o>(</span>Cytomegalovirus<span class=w> </span><span class=o>)]</span>
<span class=w>  </span>*<span class=w> </span>NC_001664.4<span class=w>   </span><span class=o>[</span>Human<span class=w> </span>betaherpesvirus<span class=w> </span>6A<span class=w> </span><span class=o>(</span>HHV-6A<span class=o>)]</span>
<span class=w>  </span>*<span class=w> </span>NC_000898.1<span class=w>   </span><span class=o>[</span>Human<span class=w> </span>betaherpesvirus<span class=w> </span>6B<span class=w> </span><span class=o>(</span>HHV-6B<span class=o>)]</span>
<span class=w>  </span>*<span class=w> </span>NC_001716.2<span class=w>   </span><span class=o>[</span>Human<span class=w> </span>betaherpesvirus<span class=w> </span><span class=m>7</span><span class=w> </span><span class=o>(</span>HHV-7<span class=o>)]</span>
<span class=w>  </span>*<span class=w> </span>NC_009333.1<span class=w>   </span><span class=o>[</span>Human<span class=w> </span>gammaherpesvirus<span class=w> </span><span class=m>8</span><span class=w> </span><span class=o>(</span>KSHV<span class=o>)]</span>
<span class=w>  </span>*<span class=w> </span>NC_045512.2<span class=w>   </span><span class=o>[</span>Severe<span class=w> </span>acute<span class=w> </span>respiratory<span class=w> </span>syndrome<span class=o>(</span>SARS<span class=o>)</span>-related<span class=w> </span>coronavirus<span class=o>]</span>
<span class=w>  </span>*<span class=w> </span>MN485971.1<span class=w>    </span><span class=o>[</span>HIV<span class=w> </span>from<span class=w> </span>Belgium<span class=o>]</span>
<span class=w>  </span>*<span class=w> </span>NC_001806.2<span class=w>   </span><span class=o>[</span>Human<span class=w> </span>alphaherpesvirus<span class=w> </span><span class=m>1</span><span class=w> </span><span class=o>(</span>Herpes<span class=w> </span>simplex<span class=w> </span>virus<span class=w> </span><span class=nb>type</span><span class=w> </span><span class=m>1</span><span class=o>)](</span>strain<span class=w> </span><span class=m>17</span><span class=o>)</span><span class=w> </span><span class=o>(</span>HSV-1<span class=o>)]</span>
<span class=w>  </span>*<span class=w> </span>KT899744.1<span class=w>    </span><span class=o>[</span>HSV-1<span class=w> </span>strain<span class=w> </span>KOS<span class=o>]</span>
<span class=w>  </span>*<span class=w> </span>MH636806.1<span class=w>    </span><span class=o>[</span>MHV68<span class=w> </span><span class=o>(</span>Murine<span class=w> </span>herpesvirus<span class=w> </span><span class=m>68</span><span class=w> </span>strain<span class=w> </span>WUMS<span class=o>)]</span>

<span class=c1>##########################################################################################</span>

USAGE:
<span class=w>  </span>bash<span class=w> </span>/data/Ziegelbauer_lab/Pipelines/circRNA/activeDev/charlie<span class=w> </span>-w/--workdir<span class=o>=</span>&lt;WORKDIR&gt;<span class=w> </span>-m/--runmode<span class=o>=</span>&lt;RUNMODE&gt;

Required<span class=w> </span>Arguments:
<span class=m>1</span>.<span class=w>  </span>WORKDIR<span class=w>     </span>:<span class=w> </span><span class=o>[</span>Type:<span class=w> </span>String<span class=o>]</span>:<span class=w> </span>Absolute<span class=w> </span>or<span class=w> </span>relative<span class=w> </span>path<span class=w> </span>to<span class=w> </span>the<span class=w> </span>output<span class=w> </span>folder<span class=w> </span>with<span class=w> </span>write<span class=w> </span>permissions.

<span class=m>2</span>.<span class=w>  </span>RUNMODE<span class=w>     </span>:<span class=w> </span><span class=o>[</span>Type:<span class=w> </span>String<span class=o>]</span><span class=w> </span>Valid<span class=w> </span>options:
<span class=w>    </span>*<span class=w> </span>init<span class=w>      </span>:<span class=w> </span>initialize<span class=w> </span>workdir
<span class=w>    </span>*<span class=w> </span>dryrun<span class=w>    </span>:<span class=w> </span>dry<span class=w> </span>run<span class=w> </span>snakemake<span class=w> </span>to<span class=w> </span>generate<span class=w> </span>DAG
<span class=w>    </span>*<span class=w> </span>run<span class=w>       </span>:<span class=w> </span>run<span class=w> </span>with<span class=w> </span>slurm
<span class=w>    </span>*<span class=w> </span>runlocal<span class=w>  </span>:<span class=w> </span>run<span class=w> </span>without<span class=w> </span>submitting<span class=w> </span>to<span class=w> </span>sbatch
<span class=w>    </span>ADVANCED<span class=w> </span>RUNMODES<span class=w> </span><span class=o>(</span>use<span class=w> </span>with<span class=w> </span>caution!!<span class=o>)</span>
<span class=w>    </span>*<span class=w> </span>unlock<span class=w>    </span>:<span class=w> </span>unlock<span class=w> </span>WORKDIR<span class=w> </span><span class=k>if</span><span class=w> </span>locked<span class=w> </span>by<span class=w> </span>snakemake<span class=w> </span>NEVER<span class=w> </span>UNLOCK<span class=w> </span>WORKDIR<span class=w> </span>WHERE<span class=w> </span>PIPELINE<span class=w> </span>IS<span class=w> </span>CURRENTLY<span class=w> </span>RUNNING!
<span class=w>    </span>*<span class=w> </span>reconfig<span class=w>  </span>:<span class=w> </span>recreate<span class=w> </span>config<span class=w> </span>file<span class=w> </span><span class=k>in</span><span class=w> </span>WORKDIR<span class=w> </span><span class=o>(</span>debugging<span class=w> </span>option<span class=o>)</span><span class=w> </span>EDITS<span class=w> </span>TO<span class=w> </span>config.yaml<span class=w> </span>WILL<span class=w> </span>BE<span class=w> </span>LOST!
<span class=w>    </span>*<span class=w> </span>reset<span class=w>     </span>:<span class=w> </span>DELETE<span class=w> </span>workdir<span class=w> </span>dir<span class=w> </span>and<span class=w> </span>re-init<span class=w> </span>it<span class=w> </span><span class=o>(</span>debugging<span class=w> </span>option<span class=o>)</span><span class=w> </span>EDITS<span class=w> </span>TO<span class=w> </span>ALL<span class=w> </span>FILES<span class=w> </span>IN<span class=w> </span>WORKDIR<span class=w> </span>WILL<span class=w> </span>BE<span class=w> </span>LOST!
<span class=w>    </span>*<span class=w> </span>printbinds:<span class=w> </span>print<span class=w> </span>singularity<span class=w> </span>binds<span class=w> </span><span class=o>(</span>paths<span class=o>)</span>
<span class=w>    </span>*<span class=w> </span><span class=nb>local</span><span class=w>     </span>:<span class=w> </span>same<span class=w> </span>as<span class=w> </span>runlocal

Optional<span class=w> </span>Arguments:

--singcache<span class=p>|</span>-c<span class=w>  </span>:<span class=w> </span>singularity<span class=w> </span>cache<span class=w> </span>directory.<span class=w> </span>Default<span class=w> </span>is<span class=w> </span><span class=sb>`</span>/data/<span class=si>${</span><span class=nv>USER</span><span class=si>}</span>/.singularity<span class=sb>`</span><span class=w> </span><span class=k>if</span><span class=w> </span>available,<span class=w> </span>or<span class=w> </span>falls<span class=w> </span>back<span class=w> </span>to<span class=w> </span><span class=sb>`</span><span class=si>${</span><span class=nv>WORKDIR</span><span class=si>}</span>/.singularity<span class=sb>`</span>.<span class=w> </span>Use<span class=w> </span>this<span class=w> </span>flag<span class=w> </span>to<span class=w> </span>specify<span class=w> </span>a<span class=w> </span>different<span class=w> </span>singularity<span class=w> </span>cache<span class=w> </span>directory.
--host<span class=p>|</span>-g<span class=w>       </span>:<span class=w> </span>supply<span class=w> </span>host<span class=w> </span>at<span class=w> </span><span class=nb>command</span><span class=w> </span>line.<span class=w> </span>hg38<span class=w> </span>or<span class=w> </span>mm39.<span class=w>                                            </span><span class=o>(</span>--runmode<span class=o>=</span>init<span class=w> </span>only<span class=o>)</span>
--additives<span class=p>|</span>-a<span class=w>  </span>:<span class=w> </span>supply<span class=w> </span>comma-separated<span class=w> </span>list<span class=w> </span>of<span class=w> </span>additives<span class=w> </span>at<span class=w> </span><span class=nb>command</span><span class=w> </span>line.<span class=w> </span>ERCC<span class=w> </span>or<span class=w> </span>BAC16Insert<span class=w> </span>or<span class=w> </span>both<span class=w> </span><span class=o>(</span>--runmode<span class=o>=</span>init<span class=w> </span>only<span class=o>)</span>
--viruses<span class=p>|</span>-v<span class=w>    </span>:<span class=w> </span>supply<span class=w> </span>comma-separated<span class=w> </span>list<span class=w> </span>of<span class=w> </span>viruses<span class=w> </span>at<span class=w> </span><span class=nb>command</span><span class=w> </span>line<span class=w>                                </span><span class=o>(</span>--runmode<span class=o>=</span>init<span class=w> </span>only<span class=o>)</span>
--manifest<span class=p>|</span>-s<span class=w>   </span>:<span class=w> </span>absolute<span class=w> </span>path<span class=w> </span>to<span class=w> </span>samples.tsv.<span class=w> </span>This<span class=w> </span>will<span class=w> </span>be<span class=w> </span>copied<span class=w> </span>to<span class=w> </span>output<span class=w> </span>folder<span class=w>                    </span><span class=o>(</span>--runmode<span class=o>=</span>init<span class=w> </span>only<span class=o>)</span>
--changegrp<span class=p>|</span>-z<span class=w>  </span>:<span class=w> </span>change<span class=w> </span>group<span class=w> </span>to<span class=w> </span><span class=s2>&quot;Ziegelbauer_lab&quot;</span><span class=w> </span>before<span class=w> </span>running<span class=w> </span>anything.<span class=w> </span>Biowulf-only.<span class=w> </span>Useful<span class=w> </span><span class=k>for</span><span class=w> </span>correctly<span class=w> </span>setting<span class=w> </span>permissions.
--help<span class=p>|</span>-h<span class=w>       </span>:<span class=w> </span>print<span class=w> </span>this<span class=w> </span><span class=nb>help</span>


Example<span class=w> </span>commands:
<span class=w>  </span>bash<span class=w> </span>/data/Ziegelbauer_lab/Pipelines/circRNA/activeDev/charlie<span class=w> </span>-w<span class=o>=</span>/my/ouput/folder<span class=w> </span>-m<span class=o>=</span>init
<span class=w>  </span>bash<span class=w> </span>/data/Ziegelbauer_lab/Pipelines/circRNA/activeDev/charlie<span class=w> </span>-w<span class=o>=</span>/my/ouput/folder<span class=w> </span>-m<span class=o>=</span>dryrun
<span class=w>  </span>bash<span class=w> </span>/data/Ziegelbauer_lab/Pipelines/circRNA/activeDev/charlie<span class=w> </span>-w<span class=o>=</span>/my/ouput/folder<span class=w> </span>-m<span class=o>=</span>run

<span class=c1>##########################################################################################</span>

VersionInfo:
<span class=w>  </span>python<span class=w>          </span>:<span class=w> </span><span class=m>3</span>.7
<span class=w>  </span>snakemake<span class=w>       </span>:<span class=w> </span><span class=m>7</span>.19.1
<span class=w>  </span>pipeline_home<span class=w>   </span>:<span class=w> </span>/vf/users/Ziegelbauer_lab/Pipelines/circRNA/activeDev
<span class=w>  </span>git<span class=w> </span>commit/tag<span class=w>  </span>:<span class=w> </span>1ae5ca091976364369784f67adffbbbf1dcdb7d5<span class=w>    </span>v0.8-197-g1ae5ca0

<span class=c1>##########################################################################################</span>
</code></pre></div> <h3 id=5-license>5. License<a class=headerlink href=#5-license title="Permanent link">&para;</a></h3> <p>MIT License</p> <p>Copyright &copy; 2021 Vishal Koparde</p> <p>Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:</p> <p>The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.</p> <p>THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.</p> <h3 id=6-testing>6. Testing<a class=headerlink href=#6-testing title="Permanent link">&para;</a></h3> <h4 id=init>Init<a class=headerlink href=#init title="Permanent link">&para;</a></h4> <p>Run init mode:</p> <div class=highlight><pre><span></span><code>bash<span class=w> </span>&lt;path<span class=w> </span>to<span class=w> </span>charlie&gt;<span class=w> </span>-w<span class=o>=</span>&lt;path<span class=w> </span>to<span class=w> </span>output<span class=w> </span>dir&gt;<span class=w> </span>-m<span class=o>=</span>init
</code></pre></div> <p>This will create the folder provided by <code>-w=</code>. The user should have write permission to this folder.</p> <h4 id=dry-run>Dry-run<a class=headerlink href=#dry-run title="Permanent link">&para;</a></h4> <p>Test data (1 paired-end subsample and 1 single-end subsample) have been including under the <code>.tests/dummy_fastqs</code> folder. After running in <code>-m=init</code>, <code>samples.tsv</code> should be edited to point the copies of the above mentioned samples with the column headers:</p> <ul> <li>sampleName</li> <li>path_to_R1_fastq</li> <li>path_to_R2_fastq</li> </ul> <p>Column <code>path_to_R2_fastq</code> will be blank in case of single-end samples.</p> <p>After editing <code>samples.tsv</code>, dry run should be run:</p> <div class=highlight><pre><span></span><code>bash<span class=w> </span>&lt;path<span class=w> </span>to<span class=w> </span>charlie&gt;<span class=w> </span>-w<span class=o>=</span>&lt;path<span class=w> </span>to<span class=w> </span>output<span class=w> </span>dir&gt;<span class=w> </span>-m<span class=o>=</span>dryrun
</code></pre></div> <p>This will create the reference fasta and gtf file based on the selections made in the <code>config.yaml</code>.</p> <h4 id=run>Run<a class=headerlink href=#run title="Permanent link">&para;</a></h4> <p>If <code>-m=dryrun</code> was sucessful, then simply do <code>-m=run</code>. The output will look something like this</p> <div class=highlight><pre><span></span><code>... ... skipping ~1000 lines
...
...
Job stats:
job                                              count    min threads    max threads
---------------------------------------------  -------  ----------
all                                                  1              1              1
annotate_clear_output                                2              1              1
circExplorer                                         2              2              2
circExplorer_bwa                                     2              2              2
circrnafinder                                        2              1              1
ciri                                                 2             56             56
clear                                                2              2              2
create_bowtie2_index                                 1              1              1
create_bwa_index                                     1              1              1
create_circExplorer_BSJ_bam                          2              4              4
create_circExplorer_linear_spliced_bams              2             56             56
create_circExplorer_merged_found_counts_table        2              1              1
create_hq_bams                                       2              1              1
create_index                                         1             56             56
create_master_counts_file                            1              1              1
cutadapt                                             2             56             56
dcc                                                  2              4              4
dcc_create_samplesheets                              2              1              1
estimate_duplication                                 2              1              1
fastqc                                               2              4              4
find_circ                                            2             56             56
find_circ_align                                      2             56             56
merge_SJ_tabs                                        1              2              2
merge_alignment_stats                                1              1              1
merge_genecounts                                     1              1              1
merge_per_sample                                     2              1              1
star1p                                               2             56             56
star2p                                               2             56             56
star_circrnafinder                                   2             56             56
total                                               52              1             56

Reasons:
    (check individual jobs above for details)
    input files updated by another job:
        alignment_stats, all, annotate_clear_output, circExplorer, circExplorer_bwa, circrnafinder, ciri, clear, create_circExplorer_BSJ_bam, create_circExplorer_linear_spliced_bams, create_circExplorer_merged_found_counts_table, create_hq_bams, create_master_counts_file, dcc, dcc_create_samplesheets, estimate_duplication, fastqc, find_circ, find_circ_align, merge_SJ_tabs, merge_alignment_stats, merge_genecounts, merge_per_sample, star1p, star2p, star_circrnafinder
    missing output files:
        alignment_stats, annotate_clear_output, circExplorer, circExplorer_bwa, circrnafinder, ciri, clear, create_bowtie2_index, create_bwa_index, create_circExplorer_BSJ_bam, create_circExplorer_linear_spliced_bams, create_circExplorer_merged_found_counts_table, create_hq_bams, create_index, create_master_counts_file, cutadapt, dcc, dcc_create_samplesheets, estimate_duplication, fastqc, find_circ, find_circ_align, merge_SJ_tabs, merge_alignment_stats, merge_genecounts, merge_per_sample, star1p, star2p, star_circrnafinder

This was a dry-run (flag -n). The order of jobs does not reflect the order of execution.
Running...
14743440
</code></pre></div> <h5 id=61-test-data>6.1 Test Data<a class=headerlink href=#61-test-data title="Permanent link">&para;</a></h5> <p>The <code>.tests/dummy_fastqs</code> folder in the repo has test dataset:</p> <div class=highlight><pre><span></span><code>%<span class=w> </span>tree<span class=w> </span>.tests/dummy_fastqs
.tests/dummy_fastqs
├──<span class=w> </span>GI1_N.R1.fastq.gz
├──<span class=w> </span>GI1_N.R2.fastq.gz
└──<span class=w> </span>GI1_T.R1.fastq.gz
</code></pre></div> <p><code>GI1_N</code> is a PE sample while <code>GI1_T</code> is a SE sample.</p> <h5 id=62-expected-output>6.2 Expected Output<a class=headerlink href=#62-expected-output title="Permanent link">&para;</a></h5> <p>Expected output from the sample data is stored under <code>.tests/expected_output</code>.</p> <p>More details about running test data can be found <a href=https://ccbr.github.io/CHARLIE/tutorial>here</a>.</p> <blockquote> <p>DISCLAIMER:</p> <p>CHARLIE is built to be run only on <a href=https://hpc.nih.gov>BIOWULF</a>. A newer HPC-agnostic version of CHARLIE is planned for 2024.</p> </blockquote> <!-- the contents are automatically inserted here. do not edit this file, instead edit the file specified above to make changes. --> <hr> <div class=md-source-file> <small> Last update: 2024-09-16 </small> </div> </article> </div> </div> <button type=button class="md-top md-icon" data-md-component=top hidden> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"/></svg> Back to top </button> </main> <footer class=md-footer> <div class="md-footer-meta md-typeset"> <div class="md-footer-meta__inner md-grid"> <div class=md-copyright> <div class=md-copyright__highlight> Copyright &copy; 2023 CCR Collaborative Bioinformatics Resource </div> Made with <a href=https://squidfunk.github.io/mkdocs-material/ target=_blank rel=noopener> Material for MkDocs </a> </div> <div class=md-social> <a href=http://bioinformatics.cancer.gov target=_blank rel=noopener title=bioinformatics.cancer.gov class=md-social__link> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 640 512"><!-- Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M144 0a80 80 0 1 1 0 160 80 80 0 1 1 0-160zm368 0a80 80 0 1 1 0 160 80 80 0 1 1 0-160zM0 298.7C0 239.8 47.8 192 106.7 192h42.7c15.9 0 31 3.5 44.6 9.7-1.3 7.2-1.9 14.7-1.9 22.3 0 38.2 16.8 72.5 43.3 96H21.3C9.6 320 0 310.4 0 298.7zM405.3 320h-.7c26.6-23.5 43.3-57.8 43.3-96 0-7.6-.7-15-1.9-22.3 13.6-6.3 28.7-9.7 44.6-9.7h42.7c58.9 0 106.7 47.8 106.7 106.7 0 11.8-9.6 21.3-21.3 21.3H405.3zM224 224a96 96 0 1 1 192 0 96 96 0 1 1-192 0zm-96 261.3c0-73.6 59.7-133.3 133.3-133.3h117.4c73.6 0 133.3 59.7 133.3 133.3 0 14.7-11.9 26.7-26.7 26.7H154.7c-14.7 0-26.7-11.9-26.7-26.7z"/></svg> </a> <a href=https://github.com/CCBR target=_blank rel=noopener title=github.com class=md-social__link> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 496 512"><!-- Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg> </a> <a href=https://hub.docker.com/orgs/nciccbr/repositories target=_blank rel=noopener title=hub.docker.com class=md-social__link> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 640 512"><!-- Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M349.9 236.3h-66.1v-59.4h66.1v59.4zm0-204.3h-66.1v60.7h66.1V32zm78.2 144.8H362v59.4h66.1v-59.4zm-156.3-72.1h-66.1v60.1h66.1v-60.1zm78.1 0h-66.1v60.1h66.1v-60.1zm276.8 100c-14.4-9.7-47.6-13.2-73.1-8.4-3.3-24-16.7-44.9-41.1-63.7l-14-9.3-9.3 14c-18.4 27.8-23.4 73.6-3.7 103.8-8.7 4.7-25.8 11.1-48.4 10.7H2.4c-8.7 50.8 5.8 116.8 44 162.1 37.1 43.9 92.7 66.2 165.4 66.2 157.4 0 273.9-72.5 328.4-204.2 21.4.4 67.6.1 91.3-45.2 1.5-2.5 6.6-13.2 8.5-17.1l-13.3-8.9zm-511.1-27.9h-66v59.4h66.1v-59.4zm78.1 0h-66.1v59.4h66.1v-59.4zm78.1 0h-66.1v59.4h66.1v-59.4zm-78.1-72.1h-66.1v60.1h66.1v-60.1z"/></svg> </a> </div> </div> </div> </footer> </div> <div class=md-dialog data-md-component=dialog> <div class="md-dialog__inner md-typeset"></div> </div> <script id=__config type=application/json>{"base": ".", "features": ["navigation.tabs", "navigation.top", "navigation.indexes", "toc.integrate"], "search": "assets/javascripts/workers/search.208ed371.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": {"provider": "mike"}}</script> <script src=assets/javascripts/bundle.51198bba.min.js></script> </body> </html>