
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
        <meta name="author" content="Vishal Koparde, Ph.D.">
      
      
      
        <link rel="prev" href="../tutorial/">
      
      
        <link rel="next" href="../versions/">
      
      
      <link rel="icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.5.3, mkdocs-material-9.5.2">
    
    
      
        <title>References - CHARLIE (Circrnas in Host And viRuses anaLysis pIpEline)</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.50c56a3b.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#charlie" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="CHARLIE (Circrnas in Host And viRuses anaLysis pIpEline)" class="md-header__button md-logo" aria-label="CHARLIE (Circrnas in Host And viRuses anaLysis pIpEline)" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            CHARLIE (Circrnas in Host And viRuses anaLysis pIpEline)
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              References
            
          </span>
        </div>
      </div>
    </div>
    
      
    
    
    
    
      <label class="md-header__button md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/CCBR/CHARLIE" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.5.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    CCBR/CHARLIE
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="CHARLIE (Circrnas in Host And viRuses anaLysis pIpEline)" class="md-nav__button md-logo" aria-label="CHARLIE (Circrnas in Host And viRuses anaLysis pIpEline)" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    CHARLIE (Circrnas in Host And viRuses anaLysis pIpEline)
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/CCBR/CHARLIE" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.5.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    CCBR/CHARLIE
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Home
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../flowchart/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Flowchart
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../tutorial/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Tutorial
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  <span class="md-ellipsis">
    References
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  <span class="md-ellipsis">
    References
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#charlie" class="md-nav__link">
    <span class="md-ellipsis">
      CHARLIE
    </span>
  </a>
  
    <nav class="md-nav" aria-label="CHARLIE">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#fasta" class="md-nav__link">
    <span class="md-ellipsis">
      Fasta
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#annotations" class="md-nav__link">
    <span class="md-ellipsis">
      Annotations
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Annotations">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#viruses" class="md-nav__link">
    <span class="md-ellipsis">
      Viruses
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../versions/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Versions
  </span>
  

      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#charlie" class="md-nav__link">
    <span class="md-ellipsis">
      CHARLIE
    </span>
  </a>
  
    <nav class="md-nav" aria-label="CHARLIE">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#fasta" class="md-nav__link">
    <span class="md-ellipsis">
      Fasta
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#annotations" class="md-nav__link">
    <span class="md-ellipsis">
      Annotations
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Annotations">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#viruses" class="md-nav__link">
    <span class="md-ellipsis">
      Viruses
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  

  
  


  <h1>References</h1>

<h2 id="charlie">CHARLIE</h2>
<p><img alt="img" src="https://img.shields.io/github/issues/kopardev/circRNA?style=for-the-badge" /><img alt="img" src="https://img.shields.io/github/forks/kopardev/circRNA?style=for-the-badge" /><img alt="img" src="https://img.shields.io/github/stars/kopardev/circRNA?style=for-the-badge" /><img alt="img" src="https://img.shields.io/github/license/kopardev/circRNA?style=for-the-badge" /></p>
<p>The reference sequences comprises of the host genome and the viral genomes.</p>
<h3 id="fasta">Fasta</h3>
<p><em>hg38</em> genome build is chosen to represent human (host). Human ribosomal sequences (<em>45S, 5S</em>) are downloaded from NCBI. <em>hg38</em> was masked for rRNA sequence and <em>45S</em> and <em>5S</em> sequences from NCBI are appended as separate chromosomes. The following viral sequences were appended to the rRNA masked <em>hg38</em> reference:</p>
<table>
<thead>
<tr>
<th>RefSeq  Sequence</th>
<th>RefSeq assembly accession</th>
<th>Notes</th>
</tr>
</thead>
<tbody>
<tr>
<td>NC_007605.1</td>
<td>GCF_002402265.1</td>
<td>Human gammaherpesvirus 4 (Epstein-Barr virus)</td>
</tr>
<tr>
<td>NC_000898.1</td>
<td>GCF_000846365.1</td>
<td>Human betaherpesvirus 6B</td>
</tr>
<tr>
<td>NC_001664.4</td>
<td>GCF_000845685.2</td>
<td>Human betaherpesvirus 6A</td>
</tr>
<tr>
<td>NC_001716.2</td>
<td>GCF_000848125.1</td>
<td>Human betaherpesvirus 7</td>
</tr>
<tr>
<td>NC_006273.2</td>
<td>GCF_000845245.1</td>
<td>Human betaherpesvirus 5</td>
</tr>
<tr>
<td>NC_009333.1</td>
<td>GCF_000838265.1</td>
<td>Human gammaherpesvirus 8</td>
</tr>
<tr>
<td>NC_045512.2</td>
<td>GCF_009858895.2</td>
<td>Severe acute respiratory syndrome-related coronavirus</td>
</tr>
<tr>
<td>MN485971.1</td>
<td>xx</td>
<td>HIV from Belgium ... GTF is hand curated</td>
</tr>
</tbody>
</table>
<p>Location: The entire resource bundle is available at <code>/data/Ziegelbauer_lab/resources/hg38_rRNA_masked_plus_rRNA_plus_viruses_plus_ERCC</code> on biowulf. This location also have additional bash scritpts required for aggregating annotations and building indices required by different aligners.</p>
<p><u><strong>Update</strong> <strong>(02/10/21)</strong></u></p>
<p>The following viral sequence has also been appended to the reference:</p>
<table>
<thead>
<tr>
<th>RefSeq  Sequence</th>
<th>RefSeq assembly accession</th>
<th>Notes</th>
</tr>
</thead>
<tbody>
<tr>
<td>NC_001806.2</td>
<td>GCF_000859985.2</td>
<td><a href="https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?mode=Info&amp;id=10298&amp;lvl=3&amp;lin=f&amp;keep=1&amp;srchmode=1&amp;unlock">Human alphaherpesvirus 1 (Herpes simplex virus type 1)</a> (strain 17)</td>
</tr>
</tbody>
</table>
<p>Location: The updated resource bundle is at <code>/data/Ziegelbauer_lab/resources/hg38_rRNA_masked_plus_rRNA_plus_viruses_plus_ERCC.v2</code> on biowulf</p>
<h3 id="annotations">Annotations</h3>
<p>Gencode release 36 is used to annotate the human transcripts. Customized RefSeq annotations are used for annotating the viral sequences. </p>
<h4 id="viruses">Viruses</h4>
<p>To properly annotate viral sequences the following process was followed:</p>
<p><strong>Downloading annotations from NCBI:</strong></p>
<p>For example, to download gtf annotations for KSHV genome search "KSHV" at <a href="https://www.ncbi.nlm.nih.gov/assembly/">NCBI</a></p>
<p><img alt="image-20200807133058218" src="https://tva1.sinaimg.cn/large/007S8ZIlgy1ghirj71jv5j30qn01taa5.jpg" /></p>
<p>On the results page (assembly page), click on "FTP directory for RefSeq assembly"</p>
<p><img alt="image-20200807133320457" src="https://tva1.sinaimg.cn/large/007S8ZIlgy1ghirln4fc7j30s60sawj6.jpg" /> </p>
<p>The FTP page has the relevant sequence and annotations files:</p>
<p><img alt="image-20200807133421026" src="https://tva1.sinaimg.cn/large/007S8ZIlgy1ghirmr2a4dj30t30ajq66.jpg" /></p>
<p>Download the file ending with <em>.gtf.gz</em>:</p>
<pre><code class="language-bash">&gt; curl -L &quot;https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/838/265/GCF_000838265.1_ViralProj14158/GCF_000838265.1_ViralProj14158_genomic.gtf.gz&quot; &gt; GCF_000838265.1_ViralProj14158_genomic.gtf.gz

&gt; gzip -d GCF_000838265.1_ViralProj14158_genomic.gtf.gz
</code></pre>
<p><strong>What needs to be fixed?</strong></p>
<p>The GTF file downloaded from NBCI (RefSeq annotations) needs fixing in order to work seemlessly with STAR and CCBR_Pipeliner:</p>
<ol>
<li>GTF is missing lines for "transcript" in column 3. Transcript coordinates can be different from "gene" coordinates. I am going to pick all non-"gene" lines for a particular gene, filter them for a particular "transcript", ~~find the leftmost and rightmost coordinate and use them as coordinates for the new "transcript" line in the new GTF file~~. <font color='blue'>Using leftmost and rightmost script eliminates the UTR regions and hence if a gene has only one transcript (most common scenario), the transcript coordinates mimic the gene coordinates.</font>
   eg. for <strong>HHV8GK18_gp01</strong>  gene these are the lines in the original file</li>
</ol>
<p><code>bash
   NC_009333.1  RefSeq  gene    4   1083    .   +   .   gene_id "HHV8GK18_gp01"; db_xref "GeneID:4961511"; gbkey "Gene"; gene "K1"; gene_biotype "protein_coding"; locus_tag "HHV8GK18_gp01";
   NC_009333.1  RefSeq  exon    29  1083    .   +   .   gene_id "HHV8GK18_gp01"; transcript_id "HHV8GK18_gp01"; gbkey "mRNA"; gene "K1"; locus_tag "HHV8GK18_gp01"; exon_number "1";
   NC_009333.1  RefSeq  CDS 105 941 .   +   0   gene_id "HHV8GK18_gp01"; transcript_id "HHV8GK18_gp01"; gbkey "CDS"; gene "K1"; locus_tag "HHV8GK18_gp01"; note "contains an Ig domain"; product "K1"; protein_id "YP_001129350.1"; exon_number "1";
   NC_009333.1  RefSeq  start_codon 105 107 .   +   0   gene_id "HHV8GK18_gp01"; transcript_id "HHV8GK18_gp01"; gbkey "CDS"; gene "K1"; locus_tag "HHV8GK18_gp01"; note "contains an Ig domain"; product "K1"; protein_id "YP_001129350.1"; exon_number "1";
   NC_009333.1  RefSeq  stop_codon  942 944 .   +   0   gene_id "HHV8GK18_gp01"; transcript_id "HHV8GK18_gp01"; gbkey "CDS"; gene "K1"; locus_tag "HHV8GK18_gp01"; note "contains an Ig domain"; product "K1"; protein_id "YP_001129350.1"; exon_number "1";</code></p>
<ol>
<li>Some transcripts have ids as "unknown_transcript_1". New name will be of the format "<strong>gene_id</strong>+_transcript_1". All child features for the transcript in question need  the <strong>transcript_id</strong> to be changed to the new name.</li>
</ol>
<p><code>bash
   NC_009333.1  RefSeq  gene    3179    17026   .   +   .   gene_id "HHV8GK18_gp03"; db_xref "GeneID:4961521"; gbkey "Gene"; gene "ORF6"; gene_biotype "protein_coding"; locus_tag "HHV8GK18_gp03";
   NC_009333.1  RefSeq  CDS 3179    6574    .   +   0   gene_id "HHV8GK18_gp03"; transcript_id "unknown_transcript_1"; gbkey "CDS"; gene "ORF6"; locus_tag "HHV8GK18_gp03"; note "herpesvirus core gene UL29 family"; product "ORF6"; protein_id "YP_001129352.1";
   NC_009333.1  RefSeq  start_codon 3179    3181    .   +   0   gene_id "HHV8GK18_gp03"; transcript_id "unknown_transcript_1"; gbkey "CDS"; gene "ORF6"; locus_tag "HHV8GK18_gp03"; note "herpesvirus core gene UL29 family"; product "ORF6"; protein_id "YP_001129352.1";
   NC_009333.1  RefSeq  stop_codon  6575    6577    .   +   0   gene_id "HHV8GK18_gp03"; transcript_id "unknown_transcript_1"; gbkey "CDS"; gene "ORF6"; locus_tag "HHV8GK18_gp03"; note "herpesvirus core gene UL29 family"; product "ORF6"; protein_id "YP_001129352.1";</code></p>
<ol>
<li>Some <strong>gene_id</strong> 's are empty. I am planning on reporting these in a separate file ending with <em>.extralines</em>. These need to be edited manually and appended to the output GTF file. I will be using "chromosome_name:start_coordinate-end_coordinate" as the format for generating a unique <strong>gene_id</strong> for replacement GTF entries. Eg.</li>
</ol>
<p><code>bash
   NC_009333.1  RefSeq  exon    118075  118097  .   -   .   gene_id ""; transcript_id "unknown_transcript_1"; gbkey "misc_RNA"; product "miR-K10"; exon_number "1";
   NC_009333.1  RefSeq  exon    127997  129368  .   +   .   gene_id ""; transcript_id "unknown_transcript_1"; gbkey "mRNA"; note "K14, ORF74; polyA_site not reported"; exon_number "1";
   NC_009333.1  RefSeq  exon    129517  130671  .   +   .   gene_id ""; transcript_id "unknown_transcript_1"; gbkey "mRNA"; note "K14, ORF74; polyA_site not reported"; exon_number "2";</code></p>
<p>will be changed to something like this:</p>
<p><code>bash
   NC_009333.1     RefSeq  gene    118075  118097  .       -       .       gene_id "NC_009333.1:118075-118097"; gene_name "NC_009333.1:118075-118097"; gene_biotype "miRNA"; gbkey "misc_RNA"; product "miR-K10";
   NC_009333.1     RefSeq  transcript      118075  118097  .       -       .       gene_id "NC_009333.1:118075-118097"; gene_name "NC_009333.1:118075-118097"; gene_biotype "miRNA"; transcript_id "NC_009333.1:118075-118097_
   transcript_1"; transcript_name "NC_009333.1:118075-118097_transcript_1"; transcript_type "miRNA"; gbkey "misc_RNA"; product "miR-K10";
   NC_009333.1     RefSeq  exon    118075  118097  .       -       .       gene_id "NC_009333.1:118075-118097"; transcript_id "NC_009333.1:118075-118097_transcript_1"; gbkey "misc_RNA"; product "miR-K10"; exon_number "1";
   NC_009333.1     RefSeq  gene    127997  130671  .       +       .       gene_id "NC_009333.1:127997_130671"; gene_name "NC_009333.1:127997_130671"; gene_biotype "mRNA"; gbkey "mRNA"; note "K14, ORF74; polyA_site not reporte
   d";
   NC_009333.1     RefSeq  transcript    127997  130671  .       +       .       gene_id "NC_009333.1:127997_130671"; gene_name "NC_009333.1:127997_130671"; gene_biotype "mRNA"; transcript_id "NC_009333.1:127997_130671_transcr
   ipt_1"; transcript_name "NC_009333.1:127997_130671_transcript_1"; transcript_type "NC_009333.1:127997_130671_transcript_1";gbkey "mRNA"; note "K14, ORF74; polyA_site not reported";
   NC_009333.1     RefSeq  exon    127997  129368  .       +       .       gene_id "NC_009333.1:127997_130671"; transcript_id "NC_009333.1:127997_130671_transcript_1"; gbkey "mRNA"; note "K14, ORF74; polyA_site not reported";
   exon_number "1";
   NC_009333.1     RefSeq  exon    129517  130671  .       +       .       gene_id "NC_009333.1:127997_130671"; transcript_id "NC_009333.1:127997_130671_transcript_1"; gbkey "mRNA"; note "K14, ORF74; polyA_site not reported";
   exon_number "2";</code></p>
<ol>
<li>Only few genes have <strong>exon</strong> features. Most of them have <strong>CDS</strong> only. The <strong>exon</strong> line is added to have coordinates same as the <strong>CDS</strong>, if it is missing.</li>
</ol>
<p>eg. This</p>
<p><code>bash
   NC_009333.1  RefSeq  gene    15756   17026   .   +   .   gene_id "HHV8GK18_gp08"; gene_name "ORF11"; gene_biotype "protein_coding"; db_xref "GeneID:4961439"; gbkey "Gene"; locus_tag "HHV8GK18_gp08";
   NC_009333.1  RefSeq  transcript  15756   16979   .   +   .   gene_id "HHV8GK18_gp08"; gene_name "ORF11"; gene_biotype "protein_coding"; transcript_id "HHV8GK18_gp08_transcript_1"; transcript_name "HHV8GK18_gp08_transcript_1"; transcript_type "protein_coding";
   NC_009333.1  RefSeq  CDS 15756   16976   .   +   0   gene_id "HHV8GK18_gp08"; transcript_id "HHV8GK18_gp08_transcript_1"; gbkey "CDS"; gene "ORF11"; locus_tag "HHV8GK18_gp08"; note "derived from herpesvirus dUTPase; related to HHV-5 UL82, UL83 and UL84"; product "ORF11"; protein_id "YP_001129357.1";
   NC_009333.1  RefSeq  start_codon 15756   15758   .   +   0   gene_id "HHV8GK18_gp08"; transcript_id "HHV8GK18_gp08_transcript_1"; gbkey "CDS"; gene "ORF11"; locus_tag "HHV8GK18_gp08"; note "derived from herpesvirus dUTPase; related to HHV-5 UL82, UL83 and UL84"; product "ORF11"; protein_id "YP_001129357.1";
   NC_009333.1  RefSeq  stop_codon  16977   16979   .   +   0   gene_id "HHV8GK18_gp08"; transcript_id "HHV8GK18_gp08_transcript_1"; gbkey "CDS"; gene "ORF11"; locus_tag "HHV8GK18_gp08"; note "derived from herpesvirus dUTPase; related to HHV-5 UL82, UL83 and UL84"; product "ORF11"; protein_id "YP_001129357.1";</code></p>
<p>becomes</p>
<p><code>bash
   NC_009333.1  RefSeq  gene    15756   17026   .   +   .   gene_id "HHV8GK18_gp08"; gene_name "ORF11"; gene_biotype "protein_coding"; db_xref "GeneID:4961439"; gbkey "Gene"; locus_tag "HHV8GK18_gp08";
   NC_009333.1  RefSeq  transcript  15756   16979   .   +   .   gene_id "HHV8GK18_gp08"; gene_name "ORF11"; gene_biotype "protein_coding"; transcript_id "HHV8GK18_gp08_transcript_1"; transcript_name "HHV8GK18_gp08_transcript_1"; transcript_type "protein_coding";
   NC_009333.1  RefSeq  exon    15756   16976   .   +   .   gene_id "HHV8GK18_gp08"; transcript_id "HHV8GK18_gp08_transcript_1"; gbkey "CDS"; gene "ORF11"; locus_tag "HHV8GK18_gp08"; note "derived from herpesvirus dUTPase; related to HHV-5 UL82, UL83 and UL84"; product "ORF11"; protein_id "YP_001129357.1";
   NC_009333.1  RefSeq  CDS 15756   16976   .   +   0   gene_id "HHV8GK18_gp08"; transcript_id "HHV8GK18_gp08_transcript_1"; gbkey "CDS"; gene "ORF11"; locus_tag "HHV8GK18_gp08"; note "derived from herpesvirus dUTPase; related to HHV-5 UL82, UL83 and UL84"; product "ORF11"; protein_id "YP_001129357.1";
   NC_009333.1  RefSeq  start_codon 15756   15758   .   +   0   gene_id "HHV8GK18_gp08"; transcript_id "HHV8GK18_gp08_transcript_1"; gbkey "CDS"; gene "ORF11"; locus_tag "HHV8GK18_gp08"; note "derived from herpesvirus dUTPase; related to HHV-5 UL82, UL83 and UL84"; product "ORF11"; protein_id "YP_001129357.1";
   NC_009333.1  RefSeq  stop_codon  16977   16979   .   +   0   gene_id "HHV8GK18_gp08"; transcript_id "HHV8GK18_gp08_transcript_1"; gbkey "CDS"; gene "ORF11"; locus_tag "HHV8GK18_gp08"; note "derived from herpesvirus dUTPase; related to HHV-5 UL82, UL83 and UL84"; product "ORF11"; protein_id "YP_001129357.1";</code></p>
<p><strong>Python script to automate:</strong></p>
<p>Simple python script take in RefSeq's GTF and spits out new GTF. This script (<em>fix_refseq_gtf.py</em>) is provided in the scripts folder of the repo.:</p>
<pre><code class="language-python"># Author: Vishal N. Koparde
# CCBR NCI
# Date: Aug, 2020


import sys,copy,argparse


parser = argparse.ArgumentParser()
parser.add_argument('-i',dest='ingtf', required=True, type=str, help=&quot;Input RefSeq GTF ..downloaded from NCBI ftp server&quot;)
parser.add_argument('-o',dest='outgtf', required=True, type=str, help=&quot;Modified Output RefSeq GTF&quot;)
args = parser.parse_args()

def get_gene_id(column9):
    x=column9.strip().split()
    for i,value in enumerate(x):
        if value==&quot;gene_id&quot;:
            gene_id_index=i+1
            break
    gene_id=x[gene_id_index]
    return gene_id

def get_gene_biotype(column9):
    x=column9.strip().split()
    found=0
    for i,value in enumerate(x):
        if value==&quot;gene_type&quot; or value==&quot;gene_biotype&quot;:
            gene_biotype_index=i+1
            found=1
            break
    if found==0:
        return '&quot;unknown&quot;;'
    gene_biotype=x[gene_biotype_index]
    return gene_biotype

def get_gene_name(column9):
    x=column9.strip().split()
    found=0
    for i,value in enumerate(x):
        if value==&quot;gene&quot; or value==&quot;gene_name&quot;:
            gene_index=i+1
            found=1
            break
    if found==0:
        return &quot;&quot;
    gene_name=x[gene_index]
    return gene_name

def get_transcript_id(column9):
    x=column9.strip().split()
    found=0
    for i,value in enumerate(x):
        if value==&quot;transcript_id&quot;:
            transcript_id_index=i+1
            found=1
            break
    if found==0:
        return '&quot;transcript_id_unknown&quot;;'
    transcript_id=x[transcript_id_index]
    return transcript_id

def fix_transcript_id(column9,g):
    x=column9.strip().split()
    found=0
    for i,value in enumerate(x):
        if value==&quot;transcript_id&quot;:
            transcript_id_index=i+1
            found=1
            break
    x[transcript_id_index]=g
    if found==0:
        x.append(&quot;transcript_id&quot;)
        x.append(g)
    x=&quot; &quot;.join(x)
    return x   

def create_new_transript_id(g,i):
    n=g.split('&quot;')
    n[-2]+=&quot;_transcript_&quot;+str(i)
    n='&quot;'.join(n)
    return n

def are_exons_present(transcript_lines):
    for l in transcript_lines:
        l_split=l.strip().split(&quot;\t&quot;)
        if l_split[2]==&quot;exon&quot;:
            return True
    else:
        return False

#create genelist
genelist=[]
gene_coords=dict()
all_gtflines=list(filter(lambda x:not x.startswith(&quot;#&quot;),open(args.ingtf).readlines()))
blank_gene_id_lines=[]
for f in all_gtflines:
    its_a_gene=0
    if f.strip().split(&quot;\t&quot;)[2]==&quot;gene&quot;:
        its_a_gene=1
    gene_id=get_gene_id(f.strip().split(&quot;\t&quot;)[8])
    if gene_id=='&quot;&quot;;':
        blank_gene_id_lines.append(f)
        continue
    genelist.append(gene_id)
    if its_a_gene==1 and not gene_id in gene_coords:
        gene_coords[gene_id]=(int(f.strip().split(&quot;\t&quot;)[3]),int(f.strip().split(&quot;\t&quot;)[4]))
genelist=list(set(genelist))
# print(genelist)
# print(len(blank_gene_id_lines))

#get genes2transcripts ... this is only for verifying that every gene has only 1 transript... this is the assumption
gene_id_2_transcript_ids=dict()
for g in genelist:
    if not g in gene_id_2_transcript_ids:
        gene_id_2_transcript_ids[g]=list()
    lines_with_gene_id=list(filter(lambda x: g in x,all_gtflines))
    non_gene_lines=list(filter(lambda x:x.split(&quot;\t&quot;)[2]!=&quot;gene&quot;,lines_with_gene_id))
    for l in non_gene_lines:
        t_id=get_transcript_id(l.strip().split(&quot;\t&quot;)[8])
        if t_id!='&quot;transcript_id_unknown&quot;;':
            gene_id_2_transcript_ids[g].append(t_id)
            gene_id_2_transcript_ids[g]=list(set(gene_id_2_transcript_ids[g]))

geneid2transcriptidfile=open(args.ingtf+&quot;.geneid2transcriptid&quot;,'w')
for k,v in gene_id_2_transcript_ids.items():
    geneid2transcriptidfile.write(&quot;%s\t%s\n&quot;%(k,v))
geneid2transcriptidfile.close()

#get genenames
gene_id_2_gene_name=dict()
for g in genelist:
    if not g in gene_id_2_gene_name:
        gene_id_2_gene_name[g]=list()
    lines_with_gene_id=list(filter(lambda x: g in x,all_gtflines))
    gene_line=list(filter(lambda x:x.split(&quot;\t&quot;)[2]==&quot;gene&quot;,lines_with_gene_id))
    # if len(gene_line)==0:
    #     for l in lines_with_gene_id:
    #         print(l,)
    gene_line=gene_line[0]
    gene_name=get_gene_name(gene_line.split(&quot;\t&quot;)[8])
    if gene_name==&quot;&quot;:
        gene_name=g
    gene_id_2_gene_name[g]=gene_name
# for k,v in gene_id_2_gene_name.items():
#     print(k,v)

#get transcript coordinates
gene_id_2_transcript_coordinates=dict()
for g in genelist:
    # print(&quot;gene=&quot;,g)
    if not g in gene_id_2_transcript_coordinates:
        gene_id_2_transcript_coordinates[g]=list()
    if len(gene_id_2_transcript_ids[g])==1:
        gene_id_2_transcript_coordinates[g].append(gene_coords[g])
    else:
        lines_with_gene_id=list(filter(lambda x: g in x,all_gtflines))
        non_gene_lines=list(filter(lambda x:x.split(&quot;\t&quot;)[2]!=&quot;gene&quot;,lines_with_gene_id))
        for t in gene_id_2_transcript_ids[g]:
            # print(&quot;transcript=&quot;,t)
            transcript_lines=list(filter(lambda x:t in x,non_gene_lines))
            coords=[]
            for l in transcript_lines:
                # print(l.strip())
                l_split=l.split(&quot;\t&quot;)
                coords.append(int(l_split[3]))
                coords.append(int(l_split[4]))
            # print()
            gene_id_2_transcript_coordinates[g].append((min(coords),max(coords)))
    # print(gene_id_2_transcript_coordinates[g])
# for k,v in gene_id_2_transcript_coordinates.items():
    # print(k,v)
# exit()

#get gene biotype\
gene_id_2_gene_biotype=dict()
for g in genelist:
    lines_with_gene_id=list(filter(lambda x: g in x,all_gtflines))
    gene_line=list(filter(lambda x:x.split(&quot;\t&quot;)[2]==&quot;gene&quot;,lines_with_gene_id))
    gene_line=gene_line[0]
    gene_biotype=get_gene_biotype(gene_line.split(&quot;\t&quot;)[8])
    gene_id_2_gene_biotype[g]=gene_biotype
# for k,v in gene_id_2_gene_biotype.items():
#     print(k,v)

out=open(args.outgtf,'w')    
for g in genelist:
    lines_with_gene_id=list(filter(lambda x: g in x,all_gtflines))
    gene_line=list(filter(lambda x:x.split(&quot;\t&quot;)[2]==&quot;gene&quot;,lines_with_gene_id))
    gene_line=gene_line[0]
    gene_line=gene_line.split(&quot;\t&quot;)
    others=gene_line.pop(-1)
    gene_line_copy=copy.copy(gene_line)
    # other key value pairs to add in the gene_line(col9)
    others_to_add=[]
    # print(&quot;others=&quot;,others)
    for o in others.strip().split(&quot;; &quot;):
        # print(&quot;o=&quot;,o)
        o2=o.split(&quot; &quot;)
        # print(&quot;o2=&quot;,o2)
        key=o2[0]
        value=o2[1:]
        value=&quot; &quot;.join(value)
        # print(&quot;key=&quot;,key)
        # print(&quot;value=&quot;,value)
        if key in [&quot;gene_id&quot;,&quot;gene&quot;,&quot;gene_name&quot;,&quot;gene_type&quot;,&quot;gene_biotype&quot;]:
            continue
        else:
            others_to_add.append(key)
            if not &quot;;&quot; in value:
                others_to_add.append(value+&quot;;&quot;)
            else:
                others_to_add.append(value)

    col9=[]
    col9.append(&quot;gene_id&quot;)
    col9.append(g)
    col9.append(&quot;gene_name&quot;)
    col9.append(gene_id_2_gene_name[g])
    col9.append(&quot;gene_biotype&quot;)
    col9.append(gene_id_2_gene_biotype[g])
    col9plus=copy.copy(col9)
    col9plus.extend(others_to_add)
    gene_col9=&quot; &quot;.join(col9plus)
    gene_line.append(gene_col9)
    gene_line=&quot;\t&quot;.join(gene_line)
    out.write(&quot;%s\n&quot;%(gene_line))

    non_gene_lines=list(filter(lambda x:x.split(&quot;\t&quot;)[2]!=&quot;gene&quot;,lines_with_gene_id))
    for i,t in enumerate(gene_id_2_transcript_ids[g]):
        transcript_line=copy.copy(gene_line_copy)
        transcript_line[2]=&quot;transcript&quot;
        transcript_line[3]=str(gene_id_2_transcript_coordinates[g][i][0])
        transcript_line[4]=str(gene_id_2_transcript_coordinates[g][i][1])
        new_trascript_id=create_new_transript_id(g,i+1)
        transcript_col9=copy.copy(col9)
        transcript_col9.append(&quot;transcript_id&quot;)
        transcript_col9.append(new_trascript_id)
        transcript_col9.append(&quot;transcript_name&quot;)
        transcript_col9.append(new_trascript_id)
        transcript_col9.append(&quot;transcript_type&quot;)
        transcript_col9.append(gene_id_2_gene_biotype[g])
        transcript_col9=&quot; &quot;.join(transcript_col9)
        transcript_line.append(transcript_col9)
        transcript_line=&quot;\t&quot;.join(transcript_line)   
        out.write(&quot;%s\n&quot;%(transcript_line))

        transcript_lines=list(filter(lambda x:t in x,non_gene_lines))
        have_exons=are_exons_present(transcript_lines)
        for l in transcript_lines:
            # print(l)
            l=l.strip().split(&quot;\t&quot;)
            tofix=l.pop(-1)
            l.append(fix_transcript_id(tofix,new_trascript_id))
            if l[2]==&quot;CDS&quot; and have_exons==False:
                l2=copy.copy(l)
                l2[7]=&quot;.&quot;
                l2[2]=&quot;exon&quot;
                l2=&quot;\t&quot;.join(l2)
                out.write(&quot;%s\n&quot;%(l2))
            l=&quot;\t&quot;.join(l)
            out.write(&quot;%s\n&quot;%(l))
            # print(l)
out.close()

out=open(args.ingtf+&quot;.extralines&quot;,'w')
for b in blank_gene_id_lines:
    out.write(b)
out.close()
</code></pre>
<p>Final versions of the GTF as located here:</p>
<p><code>/data/Ziegelbauer_lab/circRNADetection/viral_db/GTF/final_versions</code></p>












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      Copyright &copy; 2023 CCBR
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
        <div class="md-social">
  
    
    
    
    
      
      
    
    <a href="http://bioinformatics.cancer.gov" target="_blank" rel="noopener" title="bioinformatics.cancer.gov" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 640 512"><!--! Font Awesome Free 6.5.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M144 0a80 80 0 1 1 0 160 80 80 0 1 1 0-160zm368 0a80 80 0 1 1 0 160 80 80 0 1 1 0-160zM0 298.7C0 239.8 47.8 192 106.7 192h42.7c15.9 0 31 3.5 44.6 9.7-1.3 7.2-1.9 14.7-1.9 22.3 0 38.2 16.8 72.5 43.3 96H21.3C9.6 320 0 310.4 0 298.7zM405.3 320h-.7c26.6-23.5 43.3-57.8 43.3-96 0-7.6-.7-15-1.9-22.3 13.6-6.3 28.7-9.7 44.6-9.7h42.7c58.9 0 106.7 47.8 106.7 106.7 0 11.8-9.6 21.3-21.3 21.3H405.3zM224 224a96 96 0 1 1 192 0 96 96 0 1 1-192 0zm-96 261.3c0-73.6 59.7-133.3 133.3-133.3h117.4c73.6 0 133.3 59.7 133.3 133.3 0 14.7-11.9 26.7-26.7 26.7H154.7c-14.7 0-26.7-11.9-26.7-26.7z"/></svg>
    </a>
  
    
    
    
    
      
      
    
    <a href="https://github.com/CCBR" target="_blank" rel="noopener" title="github.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.5.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg>
    </a>
  
    
    
    
    
      
      
    
    <a href="https://hub.docker.com/orgs/nciccbr/repositories" target="_blank" rel="noopener" title="hub.docker.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 640 512"><!--! Font Awesome Free 6.5.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M349.9 236.3h-66.1v-59.4h66.1v59.4zm0-204.3h-66.1v60.7h66.1V32zm78.2 144.8H362v59.4h66.1v-59.4zm-156.3-72.1h-66.1v60.1h66.1v-60.1zm78.1 0h-66.1v60.1h66.1v-60.1zm276.8 100c-14.4-9.7-47.6-13.2-73.1-8.4-3.3-24-16.7-44.9-41.1-63.7l-14-9.3-9.3 14c-18.4 27.8-23.4 73.6-3.7 103.8-8.7 4.7-25.8 11.1-48.4 10.7H2.4c-8.7 50.8 5.8 116.8 44 162.1 37.1 43.9 92.7 66.2 165.4 66.2 157.4 0 273.9-72.5 328.4-204.2 21.4.4 67.6.1 91.3-45.2 1.5-2.5 6.6-13.2 8.5-17.1l-13.3-8.9zm-511.1-27.9h-66v59.4h66.1v-59.4zm78.1 0h-66.1v59.4h66.1v-59.4zm78.1 0h-66.1v59.4h66.1v-59.4zm-78.1-72.1h-66.1v60.1h66.1v-60.1z"/></svg>
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    <script id="__config" type="application/json">{"base": "..", "features": [], "search": "../assets/javascripts/workers/search.f886a092.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../assets/javascripts/bundle.d7c377c4.min.js"></script>
      
    
  </body>
</html>