<!DOCTYPE html><html class=writer-html5 lang=en> <head><meta charset=utf-8><meta http-equiv=X-UA-Compatible content="IE=edge"><meta name=viewport content="width=device-width, initial-scale=1.0"><meta name=author content="Vishal Koparde, Ph.D."><link rel="shortcut icon" href=../img/favicon.ico><title>Changelog - CHARLIE (Circrnas in Host And viRuses anaLysis pIpEline)</title><link rel=stylesheet href=../css/theme.css><link rel=stylesheet href=../css/theme_extra.css><link rel=stylesheet href=https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/styles/github.min.css><link href=../css/custom.css rel=stylesheet><link href=../css/version-select.css rel=stylesheet><script>
        // Current page data
        var mkdocs_page_name = "Changelog";
        var mkdocs_page_input_path = "CHANGELOG.md";
        var mkdocs_page_url = null;
      </script><!--[if lt IE 9]>
      <script src="../js/html5shiv.min.js"></script>
    <![endif]--><script src=https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/highlight.min.js></script><script>hljs.highlightAll();</script></head> <body class=wy-body-for-nav role=document> <div class=wy-grid-for-nav> <nav data-toggle=wy-nav-shift class="wy-nav-side stickynav"> <div class=wy-side-scroll> <div class=wy-side-nav-search> <a href=.. class="icon icon-home"> CHARLIE (Circrnas in Host And viRuses anaLysis pIpEline) </a><div role=search> <form id=rtd-search-form class=wy-form action=../search.html method=get> <input type=text name=q placeholder="Search docs" aria-label="Search docs" title="Type search term here"> </form> </div> </div> <div class="wy-menu wy-menu-vertical" data-spy=affix role=navigation aria-label="Navigation menu"> <ul> <li class=toctree-l1><a class="reference internal" href=..>Home</a> </li> </ul> <ul> <li class=toctree-l1><a href=../flowchart/ class="reference internal">Workflow</a> </li> </ul> <ul> <li class=toctree-l1><a href=../tutorial/ class="reference internal">Tutorial</a> </li> </ul> <ul> <li class=toctree-l1><a href=../references/ class="reference internal">References</a> </li> </ul> <ul class=current> <li class="toctree-l1 current"><a href=./ class="reference internal current">Changelog</a> <ul class=current> </ul> </li> </ul> <ul> <li class=toctree-l1><a href=../contributing/ class="reference internal">How to contribute</a> </li> </ul> </div> </div> </nav> <section data-toggle=wy-nav-shift class=wy-nav-content-wrap> <nav class=wy-nav-top role=navigation aria-label="Mobile navigation menu"> <i data-toggle=wy-nav-top class="fa fa-bars"></i> <a href=..>CHARLIE (Circrnas in Host And viRuses anaLysis pIpEline)</a> </nav> <div class=wy-nav-content> <div class=rst-content><div role=navigation aria-label="breadcrumbs navigation"> <ul class=wy-breadcrumbs> <li><a href=.. class="icon icon-home" aria-label=Docs></a></li> <li class="breadcrumb-item active">Changelog</li> <li class=wy-breadcrumbs-aside> <a href=https://github.com/CCBR/CHARLIE/edit/main/CHANGELOG.md>Edit on CCBR/CHARLIE</a> </li> </ul> <hr> </div> <div role=main class=document itemscope=itemscope itemtype=http://schema.org/Article> <div class=section itemprop=articleBody> <h2 id=charlie-0120>CHARLIE 0.12.0<a class=headerlink href=#charlie-0120 title="Permanent link">&para;</a></h2> <ul> <li>Now depends on ccbr_tools v0.4 for updated jobby &amp; spooker utilities. (#142, @kelly-sovacool)</li> </ul> <h2 id=charlie-0112>CHARLIE 0.11.2<a class=headerlink href=#charlie-0112 title="Permanent link">&para;</a></h2> <ul> <li>Minor documentation updates. (#138, @kelly-sovacool)</li> </ul> <h2 id=charlie-0111>CHARLIE 0.11.1<a class=headerlink href=#charlie-0111 title="Permanent link">&para;</a></h2> <ul> <li>CHARLIE was falsely throwing a file permissions error for tempdir values containing bash variables. (#118, @kelly-sovacool)</li> <li>Singularity bind paths were not being set properly. (#119, @kelly-sovacool)</li> <li>Update docker containers to set <code>$PYTHONPATH</code>. (#119, #125, @kelly-sovacool)</li> <li>Otherwise, this environment variable can be carried over and cause package conflicts when singularity is not run with <code>-C</code>.</li> <li>Also use <code>python -E</code> to ensure the <code>$PYTHONPATH</code> is not carried over. (#129, @kelly-sovacool)</li> <li>Fix <code>reconfig</code> to correctly replace variables in the config file. (#121, @kelly-sovacool)</li> <li>Prevent using excessive memory when copying reference files. (#126, @kelly-sovacool)</li> <li>Fix missing output files due to file system latency and use real (absolute) paths where possible. (#130, @kelly-sovacool)</li> <li>Update documentation to reflect biowulf usage and improved test dataset. (#132, @kelly-sovacool)</li> </ul> <h2 id=charlie-0110>CHARLIE 0.11.0<a class=headerlink href=#charlie-0110 title="Permanent link">&para;</a></h2> <ul> <li>Major updates to convert CHARLIE from a biowulf-specific to a platform-agnostic pipeline (#102, @kelly-sovacool):</li> <li>All rules now use containers instead of envmodules.</li> <li>Default config and cluster config files are provided for use on biowulf and FRCE.</li> <li>New entry <code>TEMPDIR</code> in the config file sets the temporary directory location for rules that require transient storage.</li> <li>New <code>--singcache</code> argument to provide a singularity cache dir location. The singularity cache dir is automatically set inside <code>/data/$USER/</code> or <code>$WORKDIR/</code> if <code>--singcache</code> is not provided.</li> <li>Minor documentation improvements. (#114, @kelly-sovacool)</li> </ul> <h2 id=charlie-0101>CHARLIE 0.10.1<a class=headerlink href=#charlie-0101 title="Permanent link">&para;</a></h2> <ul> <li>strand are reported together, strand from all callers are reported,</li> <li>both + and - flanking sites are reported,</li> <li>rev-comp function updated,</li> <li>updated versions of tools to match available tools on BIOWULF.</li> </ul> <h2 id=charlie-090>CHARLIE 0.9.0<a class=headerlink href=#charlie-090 title="Permanent link">&para;</a></h2> <p>Significant upgrades since the last release:</p> <ul> <li>updates to wrapper script, many new arguments/options added</li> <li>new per-sample counts table format</li> <li>new all-sample master counts matrix with min-nreads filtering and ntools column to show number of tools supporting the circRNA call</li> <li>new version of Snakemake</li> <li>cluster_status script added for forced completion of pipeline upon TIMEOUTs</li> <li>updated flowchart from lucid charts</li> <li>added circRNAfinder, find_circ, circExplorer2_bwa and other tools</li> <li>optimized execution and resource requirements</li> <li>updated viral annotations (Thanks Sara!)</li> <li>new method to extract linear counts, create linear BAMs using circExplorer2 outputs</li> <li>new job reporting using jobby and its derivatives</li> <li>separated creation of BWA and BOWTIE2 index from creation of STAR index to speed things up</li> <li>parallelized find_circ</li> <li>better cleanup (eg. deleting _STARgenome folders, etc.) for much smaller digital footprint</li> <li>multitude of comments throughout the snakefiles including listing of output file column descriptions</li> <li>preliminary GH actions added</li> </ul> <h2 id=charlie-070>CHARLIE 0.7.0<a class=headerlink href=#charlie-070 title="Permanent link">&para;</a></h2> <ul> <li>5 circRNA callers</li> <li>all-sample counts matrix with annotations</li> </ul> <h2 id=charlie-069>CHARLIE 0.6.9<a class=headerlink href=#charlie-069 title="Permanent link">&para;</a></h2> <ul> <li>Optimized pysam scripts</li> <li>fixed premature completion of singularity rules</li> </ul> <h2 id=charlie-065>CHARLIE 0.6.5<a class=headerlink href=#charlie-065 title="Permanent link">&para;</a></h2> <ul> <li>updated config.yaml to use the latest HSV-1 annotations received from Sarah (050421)</li> </ul> <h2 id=charlie-064>CHARLIE 0.6.4<a class=headerlink href=#charlie-064 title="Permanent link">&para;</a></h2> <ul> <li>create linear reads BAM file</li> <li>create linear reads BigWigs for each region in the .regions file.</li> </ul> <h2 id=charlie-063>CHARLIE 0.6.3<a class=headerlink href=#charlie-063 title="Permanent link">&para;</a></h2> <ul> <li>QOS not working for Taka... removed from cluster.json</li> <li>recall rule requires python/3.7 ... env module updated</li> </ul> <h2 id=charlie-062>CHARLIE 0.6.2<a class=headerlink href=#charlie-062 title="Permanent link">&para;</a></h2> <ul> <li>BSJ files are in BSJ subfolder... bug fix for v0.6.1</li> </ul> <h2 id=charlie-061>CHARLIE 0.6.1<a class=headerlink href=#charlie-061 title="Permanent link">&para;</a></h2> <ul> <li>customBSJs recalled from STAR alignments</li> <li>only for PE</li> <li>removes erroneously called CircExplorer BSJs</li> <li>create sense and anti-sense BSJ BAMs and BW for each reference (host+viruses)</li> <li>find reads which contribute to CIRI BSJs but not on the STAR list of BSJ reads, see if they contribute to novel (not called by STAR) BSJs and append novel BSJs to customBSJ list</li> </ul> <h2 id=charlie-060>CHARLIE 0.6.0<a class=headerlink href=#charlie-060 title="Permanent link">&para;</a></h2> <p>cutadapt_min_length to cutadapt rule... setting it to 15 in config (for miRNAs, Biot and short viral features)</p> <h2 id=charlie-050>CHARLIE 0.5.0<a class=headerlink href=#charlie-050 title="Permanent link">&para;</a></h2> <ul> <li><code>run_clear</code> is now set to True (as default)</li> <li><code>circ_quant</code> replaces <code>clear_quant</code> in the CLEAR rule. In order words, we are reusing the STAR alignment file and the circExplorer2 output file for running CLEAR. No need to run HISAT2 and TopHat (fusion-search with Bowtie1). This is much quicker.</li> <li>Using picard to estimate duplicates using<a href=https://gatk.broadinstitute.org/hc/en-us/articles/360037052812-MarkDuplicates-Picard-> <em>MarkDuplicates</em></a></li> <li>Generating a per-run <a href=https://multiqc.info/ >multiqc</a> HTML report</li> <li>Using <a href=https://www.rdocumentation.org/packages/eulerr/versions/6.1.0><em>eulerr</em></a> R package to generate CIRI-CircExplorer circRNA Venn diagrams and include them in the mulitqc report</li> <li>Gather per job cluster metadata like queue time, run time, job state etc. Stats are compiled in <strong>HPC_summary</strong> file</li> <li>CLEAR pipeline <em>quant.txt</em> file is annotated for known circRNAs</li> <li><code>WORKDIR</code> can now be a relative path</li> <li>bam2bw conversion fix for BSJ and spliced_reads. <a href=https://github.com/kopardev/circRNA/issues/17>Issue</a> closed!</li> </ul> <h2 id=charlie-040>CHARLIE 0.4.0<a class=headerlink href=#charlie-040 title="Permanent link">&para;</a></h2> <ul> <li><a href=https://github.com/YangLab/CLEAR>CLEAR</a> added.</li> <li>wrapper script (<code>run_circrna_daq.sh</code>) added for local and cluster execution.</li> <li>"spliced reads only" bam created and split by regions</li> </ul> <h2 id=charlie-030>CHARLIE 0.3.0<a class=headerlink href=#charlie-030 title="Permanent link">&para;</a></h2> <ul> <li>Lookup table for hg38 to hg19 circRNA annotations is updated... this eliminate one-to-many hits from the previous version</li> <li>BSJs extracted as different bam file.</li> <li>flowchart added</li> <li>adding slurmjobid to log/err file names</li> <li>v0.3.1 has significant (&gt;10X) performance improvements at BSJ bam creation</li> <li>v0.3.3 splits BSJ bams into human and viral bams, and also converts them to bigwigs</li> <li>v0.3.4 adds hg38_rRNA_masked_plus_rRNA_plus_viruses_plus_ERCC reference (source:Sarah)</li> </ul> <h2 id=charlie-020>CHARLIE 0.2.0<a class=headerlink href=#charlie-020 title="Permanent link">&para;</a></h2> <ul> <li>SE support added .. PE/SE samples handled concurrently</li> <li><code>envmodules</code> used in Snakemake in place of <code>module load</code> statements</li> </ul> <h2 id=charlie-010>CHARLIE 0.1.0<a class=headerlink href=#charlie-010 title="Permanent link">&para;</a></h2> <ul> <li>base version</li> <li>PE only support</li> </ul> <!-- the contents are automatically inserted here. do not edit this file, instead edit the file specified above to make changes. --> </div> </div><script async src="https://www.googletagmanager.com/gtag/js?id=G-D3SL9V30KL"></script> <script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());
  gtag('config', 'G-D3SL9V30KL'); // Replace with your tracking ID
</script> </div> </div> </section> </div> <div class=rst-versions role=note aria-label=Versions> <span class=rst-current-version data-toggle=rst-current-version> <span><a href=../references/ style="color: #fcfcfc">&laquo; Previous</a></span> <span><a href=../contributing/ style="color: #fcfcfc">Next &raquo;</a></span> </span> </div> <script src=../js/jquery-3.6.0.min.js></script> <script>var base_url = "..";</script> <script src=../js/theme_extra.js></script> <script src=../js/theme.js></script> <script src=../js/custom.js></script> <script src=../search/main.js></script> <script src=../js/version-select.js></script> <script>
        jQuery(function () {
            SphinxRtdTheme.Navigation.enable(true);
        });
    </script> </body> </html> 